<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Changelog • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Changelog" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.8.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1>Changelog <small></small></h1>
      <small>Source: <a href='https://github.com/HajkD/myTAI/blob/master/NEWS.md'><code>NEWS.md</code></a></small>
    </div>

    <div id="mytai-0-9-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-9-0" class="anchor"></a>myTAI 0.9.0<small> Unreleased </small>
</h1>
<div id="new-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions" class="anchor"></a>New Functions</h3>
<ul>
<li><p>new function <code><a href="../reference/ReverseHourglassTest.html">ReverseHourglassTest()</a></code> to perform a <code>Reverse Hourglass Test</code>. The Reverse Hourglass Test aims to statistically evaluate the existence of a reverse hourglass pattern based on TAI or TDI computations. The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) does follow an hourglass like shape. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does rather follow a reverse hourglass (low-high-low) shape.</p></li>
<li><p>new function <code><a href="../reference/reversehourglassScore.html">reversehourglassScore()</a></code> for computing the <code>Reverse Hourglass Score</code> for the <code>Reverse Hourglass Test</code></p></li>
</ul>
</div>
<div id="updated-functionality" class="section level3">
<h3 class="hasAnchor">
<a href="#updated-functionality" class="anchor"></a>Updated Functionality</h3>
<ul>
<li>function <code><a href="../reference/PlotSignature.html">PlotSignature()</a></code> receives a new <code>TestStatistic</code> (<code>TestStatistic = "ReverseHourglassTest"</code>) to perform a <code>revserse hourglass test</code> (= testing the significance of a low-high-low pattern)</li>
</ul>
</div>
</div>
    <div id="mytai-0-8-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-8-0" class="anchor"></a>myTAI 0.8.0<small> 2018-05-23 </small>
</h1>
<div id="updated-functionality-1" class="section level3">
<h3 class="hasAnchor">
<a href="#updated-functionality-1" class="anchor"></a>Updated Functionality</h3>
<ul>
<li>fix remaining issues when input is a <code>tibble</code>
</li>
</ul>
</div>
</div>
    <div id="mytai-0-7-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-7-0" class="anchor"></a>myTAI 0.7.0<small> 2018-04-11 </small>
</h1>
<div id="updates" class="section level2">
<h2 class="hasAnchor">
<a href="#updates" class="anchor"></a>Updates</h2>
<div id="new-functions-1" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-1" class="anchor"></a>New Functions</h3>
<ul>
<li>new function <code><a href="../reference/PlotCIRatio.html">PlotCIRatio()</a></code> to compute and visualize TAI/TDI etc patters using bootstrapping and confidence intervals (contributed by <a href='https://github.com/ljljolinq1010'>@ljljolinq1010</a>)</li>
</ul>
</div>
<div id="update-functionality" class="section level3">
<h3 class="hasAnchor">
<a href="#update-functionality" class="anchor"></a>Update Functionality</h3>
<ul>
<li>all functions can now handle <code>tibble</code> data as input -&gt; before there were errors thrown when input data wasn’t in strict <code>data.frame</code> format</li>
</ul>
</div>
</div>
</div>
    <div id="mytai-0-6-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-6-0" class="anchor"></a>myTAI 0.6.0<small> 2017-07-03 </small>
</h1>
<div id="updates-1" class="section level3">
<h3 class="hasAnchor">
<a href="#updates-1" class="anchor"></a>Updates</h3>
<ul>
<li><p><code><a href="../reference/is.ExpressionSet.html">is.ExpressionSet()</a></code> now prints out more detailed error messages when <a href="https://github.com/HajkD/myTAI/blob/master/vignettes/Introduction.Rmd#defining-input-data-standards">ExpressionSet format</a> is violated</p></li>
<li><p>adapt <code><a href="../reference/PlotContribution.html">PlotContribution()</a></code> to new version of <code>dplyr</code> where <code>summarise_each()</code> is deprecated.</p></li>
</ul>
<p>Error message accuring after new <code>dplyr</code> release was:</p>
<ol>
<li>Failure: <code><a href="../reference/PlotContribution.html">PlotContribution()</a></code> works properly with DivergenceExpressionSet input… (<a href='https://github.com/test-PlotContribution'>@test-PlotContribution</a>.R<a href='https://github.com/HajkD/myTAI/issues/16'>#16</a>) PlotContribution(DivergenceExpressionSetExample, legendName = “DS”) produced messages.</li>
</ol>
<p><code>summarise_each()</code> is deprecated. Use <code>summarise_all()</code>, <code>summarise_at()</code> or <code>summarise_if()</code> instead. To map <code>funs</code> over all variables, use <code>summarise_all()</code> <code>summarise_each()</code> is deprecated. Use <code>summarise_all()</code>, <code>summarise_at()</code> or <code>summarise_if()</code> instead. To map <code>funs</code> over all variables, use <code>summarise_all()</code></p>
<p>Is now fixed.</p>
</div>
</div>
    <div id="mytai-0-5-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-5-0" class="anchor"></a>myTAI 0.5.0<small> 2017-03-14 </small>
</h1>
<div id="new-functions-2" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-2" class="anchor"></a>New Functions</h3>
<ul>
<li><p>new function <code><a href="../reference/PlotSignature.html">PlotSignature()</a></code> allows users to plot evolutionary signatures across transcriptomes (based on ggplot2 -&gt; new main visualization function aiming to replace the <code><a href="../reference/PlotPattern.html">PlotPattern()</a></code> function)</p></li>
<li><p>new function <code><a href="../reference/TPI.html">TPI()</a></code> allows users to compute the Transcriptome Polymorphism Index introduced by <code>Gossmann et al., 2015</code>.</p></li>
<li><p>new function <code><a href="../reference/PlotMedians.html">PlotMedians()</a></code> allows users to compute and visualize the median expression of all age categories</p></li>
<li><p>new function <code><a href="../reference/PlotVars.html">PlotVars()</a></code> allows users to compute and visualize the expression variance of all age categories</p></li>
</ul>
</div>
<div id="updates-2" class="section level3">
<h3 class="hasAnchor">
<a href="#updates-2" class="anchor"></a>Updates</h3>
<ul>
<li><p><code><a href="../reference/PlotContribution.html">PlotContribution()</a></code> is now based on ggplot2 and loses base graphics arguments</p></li>
<li><p>now R/RcppExports.R and src/rcpp_funcs.cpp are included in the package due to previous compilation problems (see also <a href="http://stackoverflow.com/questions/34585560/travis-ci-r-package-error-in-documentation">stackoverflow discussion</a>)</p></li>
<li><p><code><a href="../reference/MatchMap.html">MatchMap()</a></code> is now based on <code><a href="http://dplyr.tidyverse.org/reference/join.html">dplyr::inner_join()</a></code> to match age category table with a gene expression dataset</p></li>
<li><p><code><a href="../reference/PlotCorrelation.html">PlotCorrelation()</a></code> has been extended and optimized for producing high publication quality plots</p></li>
<li><p><code><a href="../reference/PlotMeans.html">PlotMeans()</a></code> is now based on ggplot2 and lost all base graphics arguments.</p></li>
<li><p><code><a href="../reference/PlotRE.html">PlotRE()</a></code> is now based on ggplot2 and lost all base graphics arguments.</p></li>
</ul>
</div>
<div id="vignettes" class="section level3">
<h3 class="hasAnchor">
<a href="#vignettes" class="anchor"></a>Vignettes</h3>
<ul>
<li>In <code>Introduction</code> vignette: complete restructuring of the Introduction</li>
<li>In <code>Introduction</code> vignette: add new ggplot2 based examples</li>
</ul>
</div>
</div>
    <div id="mytai-0-4-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-4-0" class="anchor"></a>myTAI 0.4.0<small> 2016-05-04 </small>
</h1>
<div id="new-functions-3" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-3" class="anchor"></a>New Functions</h3>
<ul>
<li>a new function <code><a href="../reference/PlotSelectedAgeDistr.html">PlotSelectedAgeDistr()</a></code> allowing unsers to visualize the PS or DS gene distribution of a subset of genes stored in the input ExpressionSet object</li>
<li>a new function <code><a href="../reference/PlotGroupDiffs.html">PlotGroupDiffs()</a></code> allowing users to plot the significant differences between gene expression distributions of PS or DS groups</li>
<li>a new function <code><a href="../reference/GroupDiffs.html">GroupDiffs()</a></code> allowing users to perform statistical tests to quantify the gene expression level differences between all genes of defined PS or DS groups</li>
</ul>
</div>
<div id="updates-3" class="section level3">
<h3 class="hasAnchor">
<a href="#updates-3" class="anchor"></a>Updates</h3>
<ul>
<li><p><code><a href="../reference/PlotDistribution.html">PlotDistribution()</a></code> now uses ggplot2 to visualize the PS or DS distribution and is also based on the new function <code><a href="../reference/PlotSelectedAgeDistr.html">PlotSelectedAgeDistr()</a></code>; furthermore it loses arguments <code>plotText</code> and <code>...</code> and gains a new argument <code>legendName</code></p></li>
<li>remove arguments ‘main.text’ and ‘…’ from <code><a href="../reference/PlotCorrelation.html">PlotCorrelation()</a></code>
</li>
<li>
<code><a href="../reference/PlotCorrelation.html">PlotCorrelation()</a></code> is now based on ggplot2</li>
<li>
<code><a href="../reference/PlotGroupDiffs.html">PlotGroupDiffs()</a></code> receives a new argument <code>gene.set</code> allowing users to statistically quantify the group specific PS/DS differences of a selected set of genes</li>
<li>analogously to <code><a href="../reference/PlotGroupDiffs.html">PlotGroupDiffs()</a></code> the function <code><a href="../reference/GroupDiffs.html">GroupDiffs()</a></code> also receives a new argument <code>gene.set</code> allowing users to statistically quantify the group specific PS/DS differences of a selected set of genes</li>
<li>Fixing wrong x-axis labeling in <code><a href="../reference/PlotCategoryExpr.html">PlotCategoryExpr()</a></code> when <code>type = "stage-centered"</code> is specified</li>
<li><p><code><a href="../reference/PlotCategoryExpr.html">PlotCategoryExpr()</a></code> now also prints out the PS/DS absolute frequency distribution of the selected <code>gene.set</code></p></li>
</ul>
</div>
</div>
    <div id="mytai-0-3-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-3-0" class="anchor"></a>myTAI 0.3.0<small> 2015-07-31 </small>
</h1>
<div id="vignettes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#vignettes-1" class="anchor"></a>Vignettes</h3>
<ul>
<li>adding examples for <code><a href="../reference/PlotCategoryExpr.html">PlotCategoryExpr()</a></code> to <code>Advanced</code> Vignette</li>
<li>adding examples for <code><a href="../reference/PlotReplicateQuality.html">PlotReplicateQuality()</a></code> to <code>Expression</code> vignette</li>
</ul>
</div>
<div id="new-functions-4" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-4" class="anchor"></a>New Functions</h3>
<ul>
<li>a new function <code><a href="../reference/PlotCategoryExpr.html">PlotCategoryExpr()</a></code> allowing users to plot the expression levels of each age or divergence category as boxplot, dot plot or violin plot</li>
<li>a new function <code><a href="../reference/PlotReplicateQuality.html">PlotReplicateQuality()</a></code> allowing users to visualize the quality of biological replicates</li>
</ul>
</div>
</div>
    <div id="mytai-0-2-1" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-2-1" class="anchor"></a>myTAI 0.2.1<small> 2015-07-23 </small>
</h1>
<div id="vignettes-2" class="section level3">
<h3 class="hasAnchor">
<a href="#vignettes-2" class="anchor"></a>Vignettes</h3>
<ul>
<li>fixed a wrong example in the Enrichment vignette (<a href="https://github.com/HajkD/myTAI/commit/8d52fd60c274361dc9028dec3409abf60a738d8a" class="uri">https://github.com/HajkD/myTAI/commit/8d52fd60c274361dc9028dec3409abf60a738d8a</a>)</li>
</ul>
</div>
<div id="updates-4" class="section level3">
<h3 class="hasAnchor">
<a href="#updates-4" class="anchor"></a>Updates</h3>
<ul>
<li>
<code><a href="../reference/PlotGeneSet.html">PlotGeneSet()</a></code> and <code><a href="../reference/SelectGeneSet.html">SelectGeneSet()</a></code> now have a new argument <code>use.only.map</code> specifying whether or not instead of using a standard <code>ExpressionSet</code> a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to the function.</li>
<li>a wrong version of the edgeR Bioconductor package was imported causing version 0.2.0 to fail R CMD Check on unix based systems</li>
</ul>
</div>
</div>
    <div id="mytai-0-2-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-2-0" class="anchor"></a>myTAI 0.2.0<small> 2015-07-10 </small>
</h1>
<div id="vignettes-3" class="section level3">
<h3 class="hasAnchor">
<a href="#vignettes-3" class="anchor"></a>Vignettes</h3>
<ul>
<li><p>adding new vignette <strong>Taxonomy</strong> providing spep by step instructions on retrieving taxonomic information for any organism of interest</p></li>
<li><p>adding new vignette <strong>Expression Analysis</strong> providing use cases to perform gene expression data analysis with <code>myTAI</code></p></li>
<li><p>adding new vignette <strong>Enrichment</strong> providing step-by-step instructions on how to perform PS and DS enrichment analyses with <code><a href="../reference/PlotEnrichment.html">PlotEnrichment()</a></code></p></li>
<li><p>adding examples for <code><a href="../reference/pStrata.html">pStrata()</a></code>, <code><a href="../reference/pMatrix.html">pMatrix()</a></code>, <code><a href="../reference/pTAI.html">pTAI()</a></code>, <code><a href="../reference/pTDI.html">pTDI()</a></code>, and <code><a href="../reference/PlotContribution.html">PlotContribution()</a></code> to the <strong>Introduction</strong> Vignette</p></li>
</ul>
</div>
<div id="new-functions-5" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-5" class="anchor"></a>New Functions</h3>
<ul>
<li><p>a new function <code><a href="../reference/taxonomy.html">taxonomy()</a></code> allows users to retrieve taxonomic information for any organism of interest; this function has been taken from the <a href="https://github.com/HajkD/biomartr">biomartr</a> package and was removed from <code>biomartr</code> afterwards. Please notice, that in myTAI version 0.1.0 the Introduction vignette referenced to the <code><a href="../reference/taxonomy.html">taxonomy()</a></code> function in <code>biomartr</code>. This is no longer the case (since myTAI version 0.2.0), because now <code><a href="../reference/taxonomy.html">taxonomy()</a></code> is implemented in myTAI.</p></li>
<li><p>the new <code><a href="../reference/taxonomy.html">taxonomy()</a></code> function is based on the powerful R package <a href="https://github.com/ropensci/taxize">taxize</a>.</p></li>
<li><p>a new function <code><a href="../reference/SelectGeneSet.html">SelectGeneSet()</a></code> allows users to fastly select a subset of genes in an ExpressionSet</p></li>
<li><p>a new function <code><a href="../reference/DiffGenes.html">DiffGenes()</a></code> allows users to perform differential gene expression analysis with ExpressionSet objects</p></li>
<li><p>a new function <code><a href="../reference/EnrichmentTest.html">EnrichmentTest()</a></code> allows users to perform a Fisher’s exact test based enrichment analysis of over or underrepresented Phylostrata or Divergence Strata within a given gene set without having to plot the result</p></li>
<li><p>a new function <code><a href="../reference/PlotGeneSet.html">PlotGeneSet()</a></code> allows users to visualize the expression profiles of a given gene set</p></li>
<li><p>a new function <code><a href="../reference/PlotEnrichment.html">PlotEnrichment()</a></code> allows users to visualize the Phylostratum or Divergence Stratum enrichment of a given Gene Set as well as computing Fisher’s exact test to quantify the statistical significance of enrichment</p></li>
<li><p>a new function <code><a href="../reference/PlotContribution.html">PlotContribution()</a></code> allows users to visualize the Phylostratum or Divergence Stratum contribution to the global TAI/TDI pattern</p></li>
<li><p>a new function <code><a href="../reference/pTAI.html">pTAI()</a></code> allows users to compute the phylostratum contribution to the global TAI pattern</p></li>
<li><p>a new function <code><a href="../reference/pTDI.html">pTDI()</a></code> allows users to compute the divergence stratum contribution to the global TDI pattern</p></li>
</ul>
</div>
<div id="updates-5" class="section level3">
<h3 class="hasAnchor">
<a href="#updates-5" class="anchor"></a>Updates</h3>
<ul>
<li>
<code>FilterRNASeqCT()</code> has been renamed to <code><a href="../reference/Expressed.html">Expressed()</a></code> allowing users to apply this filter function to RNA-Seq data as well as to microarray data</li>
<li>
<code><a href="../reference/PlotRE.html">PlotRE()</a></code> and <code><a href="../reference/PlotMeans.html">PlotMeans()</a></code> are now based on colors from the RColorBrewer package (default)</li>
<li>
<code><a href="../reference/PlotRE.html">PlotRE()</a></code> and <code><a href="../reference/PlotMeans.html">PlotMeans()</a></code> now have a new argument <code>colors</code> allowing unsers to choose custom colors for the visualized relative or mean expression profiles</li>
<li>
<code><a href="../reference/geom.mean.html">geom.mean()</a></code> and <code><a href="../reference/harm.mean.html">harm.mean()</a></code> now are external functions accessible to the <code>myTAI</code> user</li>
</ul>
</div>
</div>
    <div id="mytai-0-1-0" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-1-0" class="anchor"></a>myTAI 0.1.0<small> 2015-05-24 </small>
</h1>
<div id="main-news" class="section level3">
<h3 class="hasAnchor">
<a href="#main-news" class="anchor"></a>Main News</h3>
<ul>
<li>now all functions have unit tests</li>
</ul>
</div>
<div id="new-functions-6" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-6" class="anchor"></a>New Functions</h3>
<ul>
<li><p>a new function <code><a href="../reference/pStrata.html">pStrata()</a></code> allows users to compute partial TAI/TDI values for all Phylostrata or Divergence Strata</p></li>
<li><p>a new function <code><a href="../reference/CollapseReplicates.html">CollapseReplicates()</a></code> allows users to combine replicate expression levels in ExpressionSet objects</p></li>
<li><p>a new function <code>FilterRNASeqCT()</code> allows users to filter expression levels of <code>ExpressionSet</code> objects deriving from RNA-Seq count tables</p></li>
</ul>
</div>
<div id="updates-6" class="section level3">
<h3 class="hasAnchor">
<a href="#updates-6" class="anchor"></a>Updates</h3>
<ul>
<li><p>function <code><a href="../reference/MatchMap.html">MatchMap()</a></code> now receives a new argument <code>remove.duplicates</code> allowing users to delete duplicate gene ids (that might be stored in the input PhyoMap or DivergenceMap) during the process of matching a Map with an ExpressionSet</p></li>
<li><p><code><a href="../reference/FlatLineTest.html">FlatLineTest()</a></code>, <code><a href="../reference/ReductiveHourglassTest.html">ReductiveHourglassTest()</a></code>, <code><a href="../reference/EarlyConservationTest.html">EarlyConservationTest()</a></code>, and <code><a href="../reference/PlotPattern.html">PlotPattern()</a></code> implement a new argument <code>custom.perm.matrix</code> allowing users to pass their own (custom) permutation matrix to the corresponding function. All subsequent test statistics and p-value/std.dev computations are then based on this custom permutation matrix</p></li>
<li><p><code><a href="../reference/EarlyConservationTest.html">EarlyConservationTest()</a></code> and <code><a href="../reference/ReductiveHourglassTest.html">ReductiveHourglassTest()</a></code> now have a new parameter <code>gof.warning</code> allowing users to choose whether or not non significant goodness of fit results should be printed as warning</p></li>
<li><p>now when specifying <code>TestStatistic = NULL</code> in <code><a href="../reference/PlotPattern.html">PlotPattern()</a></code> only the TAI/TDI profile is drawn (without performing any test statistics); this is equavalent to performing: <code>plot(TAI(PhyloExpressionSetExample)</code></p></li>
<li><p>function <code>combinatorialSignificance()</code> is now named <code><a href="../reference/CombinatorialSignificance.html">CombinatorialSignificance()</a></code></p></li>
<li><p>changing the title and description of the <code>myTAI</code> package</p></li>
<li><p>some minor changes in vignettes and within the documentation of functions</p></li>
</ul>
</div>
</div>
    <div id="mytai-0-0-2" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-0-2" class="anchor"></a>myTAI 0.0.2<small> 2014-12-27 </small>
</h1>
<div id="new-features-in-v--0-0-2" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-in-v--0-0-2" class="anchor"></a>New Features in v. 0.0.2</h2>
<ul>
<li><p><code>combinatorialSignificance()</code>, <code><a href="../reference/FlatLineTest.html">FlatLineTest()</a></code>, <code><a href="../reference/ReductiveHourglassTest.html">ReductiveHourglassTest()</a></code>, and <code><a href="../reference/EarlyConservationTest.html">EarlyConservationTest()</a></code> now support multicore processing</p></li>
<li><p><code><a href="../reference/MatchMap.html">MatchMap()</a></code> has been entirely rewritten and is now based on <a href="https://github.com/hadley/dplyr">dplyr</a>; additionally it now has a new argument <code>accumulate</code> that allows you to accumulate multiple expression levels to a unique expressiion level for a unique gene id</p></li>
</ul>
<div id="vignettes-4" class="section level3">
<h3 class="hasAnchor">
<a href="#vignettes-4" class="anchor"></a>Vignettes</h3>
<p>All three Vignettes: <code>Introduction</code>, <code>Intermediate</code>, and <code>Advanced</code> have been updated and extended.</p>
</div>
<div id="bug-fixes" class="section level3">
<h3 class="hasAnchor">
<a href="#bug-fixes" class="anchor"></a>Bug Fixes</h3>
<ul>
<li><p>two small bugs in <code><a href="../reference/ReductiveHourglassTest.html">ReductiveHourglassTest()</a></code> and <code><a href="../reference/EarlyConservationTest.html">EarlyConservationTest()</a></code> have been fixed that caused that instead of displaying 3 or 4 plots (<code><a href="https://www.rdocumentation.org/packages/graphics/topics/par">par(mfrow=c(1,3))</a></code> or <code><a href="https://www.rdocumentation.org/packages/graphics/topics/par">par(mfrow=c(2,2))</a></code>) only 1 plot has been generated</p></li>
<li><p>a small bug in <code><a href="../reference/PlotMeans.html">PlotMeans()</a></code> that caused the visualization of a wrong y-axis label when plotting only one group of Phylostrata or Divergence Strata</p></li>
</ul>
</div>
</div>
</div>
    <div id="mytai-0-0-1" class="section level1">
<h1 class="page-header">
<a href="#mytai-0-0-1" class="anchor"></a>myTAI 0.0.1<small> 2014-11-01 </small>
</h1>
<p>Introducing myTAI 0.0.1:</p>
<p>A framework to perform phylotranscriptomics analyses for Evolutionary Developmental Biology research.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
        <li><a href="#mytai-0-9-0">0.9.0</a></li>
        <li><a href="#mytai-0-8-0">0.8.0</a></li>
        <li><a href="#mytai-0-7-0">0.7.0</a></li>
        <li><a href="#mytai-0-6-0">0.6.0</a></li>
        <li><a href="#mytai-0-5-0">0.5.0</a></li>
        <li><a href="#mytai-0-4-0">0.4.0</a></li>
        <li><a href="#mytai-0-3-0">0.3.0</a></li>
        <li><a href="#mytai-0-2-1">0.2.1</a></li>
        <li><a href="#mytai-0-2-0">0.2.0</a></li>
        <li><a href="#mytai-0-1-0">0.1.0</a></li>
        <li><a href="#mytai-0-0-2">0.0.2</a></li>
        <li><a href="#mytai-0-0-1">0.0.1</a></li>
      </ul>
    </div>
  </div>

</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.2.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

