<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Evolutionary Transcriptomics • myTAI</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Evolutionary Transcriptomics">
<meta property="og:description" content="Investigate the evolution of biological processes by capturing evolutionary signatures in transcriptomes (Drost et al. (2017) &lt;doi:10.1093/bioinformatics/btx835&gt;). The aim of this tool is to provide a transcriptome analysis environment for answering questions regarding the evolution of biological processes (Drost et al. (2016) &lt;doi:10.1101/051565&gt;).">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
<div id="mytai" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#mytai" class="anchor"></a>myTAI <img src="inst/myTAI.png" align="right" height="174" width="150">
</h1></div>

<div id="evolutionary-transcriptomics-with-r" class="section level3">
<h3 class="hasAnchor">
<a href="#evolutionary-transcriptomics-with-r" class="anchor"></a>Evolutionary Transcriptomics with R</h3>
<p>Today, phenotypic phenomena such as morphological mutations, diseases or developmental processes are primarily investigated on the molecular level using transcriptomics approaches. Transcriptomes denote the total number of quantifiable transcripts present at a specific stage in a biological process. In disease or developmental (defect) studies transcriptomes are usually measured over several time points. In treatment studies aiming to quantify differences in the transcriptome due to biotic stimuli, abiotic stimuli, or diseases usually treatment / disease versus non-treatment / non-disease transcriptomes are being compared. In either case, comparing changes in transcriptomes over time or between treatments allows us to identify genes and gene regulatory mechanisms that might be involved in governing the biological process of investigation. Although transcriptomics studies are based on a powerful methodology little is known about the evolution of such transcriptomes. Understanding the evolutionary mechanism that change transcriptomes over time, however, might give us a new perspective on how diseases emerge in the first place or how morphological changes are triggered by changes of developmental transcriptomes.</p>
<p>Evolutionary transcriptomics aims to capture and quantify the evolutionary conservation of genes that contribute to the transcriptome during a specific stage of the biological process of interest. The resulting temporal conservation pattern enables then to detect stages of development or other biological processes that are evolutionarily constrained (<a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btx835/4772684">Drost et al., 2018</a>). This quantification on the highest level is achieved through transcriptome indices (e.g. <a href="https://hajkd.github.io/myTAI/articles/Introduction.html#transcriptome-age-index">Transcriptome Age Index or Transcriptome Divergence Index</a>) which aim to quantify the average evolutionary age or sequence conseration of genes that contribute to the transcriptome at a particular stage. In general, evolutionary transcriptomics can be used as a method to quantify the evolutionary conservation of transcriptomes to investigate how transcriptomes underlying biological processes are constrained or channeled due to evolutionary history (Dollow’s law) (<a href="http://www.sciencedirect.com/science/article/pii/S0959437X16302040">Drost et al., 2017</a>).</p>
<p>In principle, any transcriptome dataset published so far can be combined with evolutionary information. Thus, <code>myTAI</code> in combination with evolutionary information can be used to study corresponding transcriptomes with any available transcriptome dataset.</p>
<p>For the purpose of performing large scale evolutionary transcriptomics studies, the <code>myTAI</code> package implements frameworks to allow researchers to study the evolution of biological processes and to detect stages or periods of evolutionary conservation or variability.</p>
<p>I hope that <code>myTAI</code> will become the community standard tool to perform evolutionary transcriptomics studies and I am happy to add required functionality upon request.</p>
<p><strong>Please note, since myTAI relies on gene age inference and there has been an extensive debate about the best approaches for gene age inference in the last years, please follow my <a href="https://hajkd.github.io/myTAI/articles/Introduction.html#genomic-phylostratography-based-gene-age-inference">updated discussion about the gene age inference literature</a>.</strong></p>
<p>The following tutorials will provide use cases and detailed explainations of how to quantify transcriptome onservation with <code>myTAI</code> and how to interpret the results generated with this software tool.</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Please cite the following paper when using <code>myTAI</code> for your own research. This will allow me to continue working on this software tool and will motivate me to extend its functionality and usability in the next years. Many thanks in advance :)</p>
<blockquote>
<p>Drost et al. <strong>myTAI: evolutionary transcriptomics with R</strong> . <em>Bioinformatics</em> 2018, 34 (9), 1589-1590. <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btx835/4772684">doi:10.1093</a></p>
</blockquote>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Users can download <code>myTAI</code> from CRAN :</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install myTAI 0.8.0 from CRAN</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/source">source</a></span>(<span class="st">"http://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">'myTAI'</span>)</code></pre></div>
</div>
<div id="install-developer-version" class="section level2">
<h2 class="hasAnchor">
<a href="#install-developer-version" class="anchor"></a>Install Developer Version</h2>
<p>Some bug fixes or new functionality will not be available on CRAN yet, but in the developer version here on GitHub. To download and install the most recent version of <code>myTAI</code> run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install the developer version of myTAI on your system</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/source">source</a></span>(<span class="st">"http://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"HajkD/myTAI"</span>)</code></pre></div>
</div>
<div id="news" class="section level2">
<h2 class="hasAnchor">
<a href="#news" class="anchor"></a>NEWS</h2>
<p>The current status of the package as well as a detailed history of the functionality of each version of <code>myTAI</code> can be found in the <a href="https://hajkd.github.io/myTAI/news/index.html">NEWS</a> section.</p>
</div>
<div id="tutorials" class="section level2">
<h2 class="hasAnchor">
<a href="#tutorials" class="anchor"></a>Tutorials</h2>
<p>These tutorials introduce users to <code>myTAI</code>:</p>
<ul>
<li><a href="https://hajkd.github.io/myTAI/articles/Introduction.html">Introduction to the myTAI Package</a></li>
<li><a href="https://hajkd.github.io/myTAI/articles/Intermediate.html">Intermediate Concepts of Phylotranscriptomics</a></li>
<li><a href="https://hajkd.github.io/myTAI/articles/Advanced.html">Advanced Topics of Phylotranscriptomics</a></li>
<li><a href="https://hajkd.github.io/myTAI/articles/Enrichment.html">Perform Age Enrichment Analyses</a></li>
<li><a href="https://hajkd.github.io/myTAI/articles/Expression.html">Gene Expression Analysis with myTAI</a></li>
<li><a href="https://hajkd.github.io/myTAI/articles/Taxonomy.html">Taxonomic Information Retrieval</a></li>
</ul>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(myTAI)
<span class="co"># example dataset covering 7 stages of A thaliana embryo development</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(<span class="st">"PhyloExpressionSetExample"</span>)
<span class="co"># transform absolute expression levels to log2 expression levels</span>
ExprExample &lt;-<span class="st"> </span><span class="kw"><a href="reference/tf.html">tf</a></span>(PhyloExpressionSetExample, log2)
<span class="co"># visualize global Transcriptome Age Index pattern</span>
<span class="kw"><a href="reference/PlotSignature.html">PlotSignature</a></span>(ExprExample)
<span class="co"># plot expression level distributions for each age (=PS) category </span>
<span class="co"># and each developmental stage </span>
<span class="kw"><a href="reference/PlotCategoryExpr.html">PlotCategoryExpr</a></span>(ExprExample, <span class="st">"PS"</span>)
<span class="co"># plot median expression of each age category seperated by old (PS1-3)</span>
<span class="co"># versus young (PS4-12) genes</span>
<span class="kw"><a href="reference/PlotMedians.html">PlotMedians</a></span>(ExprExample, <span class="dt">Groups =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">4</span><span class="op">:</span><span class="dv">12</span>))
<span class="co"># plot mean expression of each age category seperated by old (PS1-3)</span>
<span class="co"># versus young (PS4-12) genes</span>
<span class="kw"><a href="reference/PlotMeans.html">PlotMeans</a></span>(ExprExample, <span class="dt">Groups =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">4</span><span class="op">:</span><span class="dv">12</span>))
<span class="co"># plot relative mean expression of each age category seperated by old (PS1-3)</span>
<span class="co"># versus young (PS4-12) genes</span>
<span class="kw"><a href="reference/PlotRE.html">PlotRE</a></span>(ExprExample, <span class="dt">Groups =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">4</span><span class="op">:</span><span class="dv">12</span>))
<span class="co"># plot the significant differences between gene expression distributions </span>
<span class="co"># of old (=group1) versus young (=group2) genes</span>
<span class="kw"><a href="reference/PlotGroupDiffs.html">PlotGroupDiffs</a></span>(<span class="dt">ExpressionSet =</span> ExprExample,
               <span class="dt">Groups        =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">group_1 =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">group_2 =</span> <span class="dv">4</span><span class="op">:</span><span class="dv">12</span>),
               <span class="dt">legendName    =</span> <span class="st">"PS"</span>,
               <span class="dt">plot.type     =</span> <span class="st">"boxplot"</span>)</code></pre></div>
<div id="package-dependencies" class="section level3">
<h3 class="hasAnchor">
<a href="#package-dependencies" class="anchor"></a>Package Dependencies</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># to perform differential gene expression analyses with myTAI</span>
<span class="co"># please install the edgeR package</span>
<span class="co"># install edgeR</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/source">source</a></span>(<span class="st">"http://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"edgeR"</span>)</code></pre></div>
</div>
</div>
<div id="getting-started-with-mytai" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started-with-mytai" class="anchor"></a>Getting started with <code>myTAI</code>
</h2>
<p>Users can also read the tutorials within (<a href="http://www.rstudio.com/">RStudio</a>) :</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># source the myTAI package</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(myTAI)

<span class="co"># look for all tutorials (vignettes) available in the myTAI package</span>
<span class="co"># this will open your web browser</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/browseVignettes">browseVignettes</a></span>(<span class="st">"myTAI"</span>)

<span class="co"># or as single tutorials</span>

<span class="co"># open tutorial: Introduction to Phylotranscriptomics and myTAI</span>
 <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/vignette">vignette</a></span>(<span class="st">"Introduction"</span>, <span class="dt">package =</span> <span class="st">"myTAI"</span>)

<span class="co"># open tutorial: Intermediate Concepts of Phylotranscriptomics</span>
 <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/vignette">vignette</a></span>(<span class="st">"Intermediate"</span>, <span class="dt">package =</span> <span class="st">"myTAI"</span>)

<span class="co"># open tutorial: Advanced Concepts of Phylotranscriptomics</span>
 <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/vignette">vignette</a></span>(<span class="st">"Advanced"</span>, <span class="dt">package =</span> <span class="st">"myTAI"</span>)

<span class="co"># open tutorial: Age Enrichment Analyses</span>
 <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/vignette">vignette</a></span>(<span class="st">"Enrichment"</span>, <span class="dt">package =</span> <span class="st">"myTAI"</span>)
 
<span class="co"># open tutorial: Gene Expression Analysis with myTAI</span>
 <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/vignette">vignette</a></span>(<span class="st">"Expression"</span>, <span class="dt">package =</span> <span class="st">"myTAI"</span>)
 
 <span class="co"># open tutorial: Taxonomic Information Retrieval with myTAI</span>
 <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/vignette">vignette</a></span>(<span class="st">"Taxonomy"</span>, <span class="dt">package =</span> <span class="st">"myTAI"</span>)</code></pre></div>
<p>In the <code>myTAI</code> framework users can find:</p>
<div id="phylotranscriptomics-measures" class="section level4">
<h4 class="hasAnchor">
<a href="#phylotranscriptomics-measures" class="anchor"></a>Phylotranscriptomics Measures:</h4>
<ul>
<li>
<code><a href="reference/TAI.html">TAI()</a></code> : Function to compute the Transcriptome Age Index (TAI)</li>
<li>
<code><a href="reference/TDI.html">TDI()</a></code> : Function to compute the Transcriptome Divergence Index (TDI)</li>
<li>
<code><a href="reference/TPI.html">TPI()</a></code> : Function to compute the Transcriptome Polymorphism Index (TPI)</li>
<li>
<code><a href="reference/REMatrix.html">REMatrix()</a></code> : Function to compute the relative expression profiles of all phylostrata or divergence-strata</li>
<li>
<code><a href="reference/RE.html">RE()</a></code> : Function to transform mean expression levels to relative expression levels</li>
<li>
<code><a href="reference/pTAI.html">pTAI()</a></code> : Compute the Phylostratum Contribution to the global TAI</li>
<li>
<code><a href="reference/pTDI.html">pTDI()</a></code> : Compute the Divergence Stratum Contribution to the global TDI</li>
<li>
<code><a href="reference/pMatrix.html">pMatrix()</a></code> : Compute Partial TAI or TDI Values</li>
<li>
<code><a href="reference/pStrata.html">pStrata()</a></code> : Compute Partial Strata Values</li>
</ul>
</div>
<div id="visualization-and-analytics-tools" class="section level4">
<h4 class="hasAnchor">
<a href="#visualization-and-analytics-tools" class="anchor"></a>Visualization and Analytics Tools:</h4>
<ul>
<li>
<code><a href="reference/PlotSignature.html">PlotSignature()</a></code> : Main visualization function to plot evolutionary signatures across transcriptomes</li>
<li>
<code><a href="reference/PlotPattern.html">PlotPattern()</a></code> : Base graphics function to plot evolutionary signatures across transcriptomes</li>
<li>
<code><a href="reference/PlotContribution.html">PlotContribution()</a></code> : Plot Cumuative Transcriptome Index</li>
<li>
<code><a href="reference/PlotCorrelation.html">PlotCorrelation()</a></code> : Function to plot the correlation between phylostratum values and divergence-stratum values</li>
<li>
<code><a href="reference/PlotRE.html">PlotRE()</a></code> : Function to plot the relative expression profiles</li>
<li>
<code><a href="reference/PlotBarRE.html">PlotBarRE()</a></code> : Function to plot the mean relative expression levels of phylostratum or divergence-stratum classes as barplot</li>
<li>
<code><a href="reference/PlotMeans.html">PlotMeans()</a></code> : Function to plot the mean expression profiles of age categories</li>
<li>
<code><a href="reference/PlotMedians.html">PlotMedians()</a></code> : Function to plot the median expression profiles of age categories</li>
<li>
<code><a href="reference/PlotVars.html">PlotVars()</a></code> : Function to plot the expression variance profiles of age categories</li>
<li>
<code><a href="reference/PlotDistribution.html">PlotDistribution()</a></code> : Function to plot the frequency distribution of genes within the corresponding age categories</li>
<li>
<code><a href="reference/PlotCategoryExpr.html">PlotCategoryExpr()</a></code> : Plot the Expression Levels of each Age or Divergence Category as Barplot or Violinplot</li>
<li>
<code><a href="reference/PlotEnrichment.html">PlotEnrichment()</a></code> : Plot the Phylostratum or Divergence Stratum Enrichment of a given Gene Set</li>
<li>
<code><a href="reference/PlotGeneSet.html">PlotGeneSet()</a></code> : Plot the Expression Profiles of a Gene Set</li>
<li>
<code><a href="reference/PlotGroupDiffs.html">PlotGroupDiffs()</a></code> : Plot the significant differences between gene expression distributions of PS or DS groups</li>
<li>
<code><a href="reference/PlotSelectedAgeDistr.html">PlotSelectedAgeDistr()</a></code> : Plot the PS or DS distribution of a selected set of genes</li>
</ul>
</div>
<div id="a-statistical-framework-and-test-statistics" class="section level4">
<h4 class="hasAnchor">
<a href="#a-statistical-framework-and-test-statistics" class="anchor"></a>A Statistical Framework and Test Statistics:</h4>
<ul>
<li>
<code><a href="reference/FlatLineTest.html">FlatLineTest()</a></code> : Function to perform the <strong>Flat Line Test</strong> that quantifies the statistical significance of an observed phylotranscriptomics pattern (significant deviation from a frat line = evolutionary signal)</li>
<li>
<code><a href="reference/ReductiveHourglassTest.html">ReductiveHourglassTest()</a></code> : Function to perform the <strong>Reductive Hourglass Test</strong> that statistically evaluates the existence of a phylotranscriptomic hourglass pattern (hourglass model)</li>
<li>
<code><a href="reference/EarlyConservationTest.html">EarlyConservationTest()</a></code> : Function to perform the <strong>Reductive Early Conservation Test</strong> that statistically evaluates the existence of a monotonically increasing phylotranscriptomic pattern (early conservation model)</li>
<li>
<code>ReverseHourglassTest</code>: Function to perform the <strong>Reverse Hourglass Test</strong> that statistically evaluates the existence of a reverse hourglass pattern (low-high-low)</li>
<li>
<code><a href="reference/EnrichmentTest.html">EnrichmentTest()</a></code> : Phylostratum or Divergence Stratum Enrichment of a given Gene Set based on Fisher’s Test</li>
<li>
<code><a href="reference/bootMatrix.html">bootMatrix()</a></code> : Compute a Permutation Matrix for Test Statistics</li>
</ul>
<p>All functions also include visual analytics tools to quantify the goodness of test statistics.</p>
</div>
<div id="differential-gene-expression-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#differential-gene-expression-analysis" class="anchor"></a>Differential Gene Expression Analysis</h4>
<ul>
<li>
<code><a href="reference/DiffGenes.html">DiffGenes()</a></code> : Implements Popular Methods for Differential Gene Expression Analysis</li>
<li>
<code><a href="reference/CollapseReplicates.html">CollapseReplicates()</a></code> : Combine Replicates in an ExpressionSet</li>
<li>
<code><a href="reference/CombinatorialSignificance.html">CombinatorialSignificance()</a></code> : Compute the Statistical Significance of Each Replicate Combination</li>
<li>
<code><a href="reference/Expressed.html">Expressed()</a></code> : Filter Expression Levels in Gene Expression Matrices (define expressed genes)</li>
<li>
<code><a href="reference/SelectGeneSet.html">SelectGeneSet()</a></code> : Select a Subset of Genes in an ExpressionSet</li>
<li>
<code><a href="reference/PlotReplicateQuality.html">PlotReplicateQuality()</a></code> : Plot the Quality of Biological Replicates</li>
<li>
<code><a href="reference/GroupDiffs.html">GroupDiffs()</a></code> : Quantify the significant differences between gene expression distributions of PS or DS groups</li>
</ul>
</div>
<div id="taxonomic-information-retrieval" class="section level4">
<h4 class="hasAnchor">
<a href="#taxonomic-information-retrieval" class="anchor"></a>Taxonomic Information Retrieval</h4>
<ul>
<li>
<code><a href="reference/taxonomy.html">taxonomy()</a></code> : Retrieve Taxonomic Information for any Organism of Interest</li>
</ul>
</div>
<div id="minor-functions-for-better-usibility-and-additional-analyses" class="section level4">
<h4 class="hasAnchor">
<a href="#minor-functions-for-better-usibility-and-additional-analyses" class="anchor"></a>Minor Functions for Better Usibility and Additional Analyses</h4>
<ul>
<li>
<code><a href="reference/MatchMap.html">MatchMap()</a></code> : Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix</li>
<li>
<code><a href="reference/tf.html">tf()</a></code> : Transform Gene Expression Levels</li>
<li>
<code><a href="reference/age.apply.html">age.apply()</a></code> : Age Category Specific apply Function</li>
<li>
<code><a href="reference/ecScore.html">ecScore()</a></code> : Compute the Hourglass Score for the EarlyConservationTest</li>
<li>
<code><a href="reference/geom.mean.html">geom.mean()</a></code> : Geometric Mean</li>
<li>
<code><a href="reference/harm.mean.html">harm.mean()</a></code> : Harmonic Mean</li>
<li>
<code><a href="reference/omitMatrix.html">omitMatrix()</a></code> : Compute TAI or TDI Profiles Omitting a Given Gene</li>
<li>
<code><a href="reference/rhScore.html">rhScore()</a></code> : Compute the Hourglass Score for the Reductive Hourglass Test</li>
<li>
<code><a href="reference/reversehourglassScore.html">reversehourglassScore()</a></code>: Compute the Reverse Hourglass Score for the Reverse Hourglass Test</li>
</ul>
</div>
</div>
<div id="developer-version-of-mytai" class="section level2">
<h2 class="hasAnchor">
<a href="#developer-version-of-mytai" class="anchor"></a>Developer Version of <code>myTAI</code>
</h2>
<p>The developer version of <code>myTAI</code> might include more functionality than the stable version on CRAN. Hence users can download the current developer version of <code>myTAI</code> by typing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># The developer version can be installed directly from github:</span>

<span class="co"># install.packages("devtools")</span>

<span class="co"># install developer version of myTAI</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(devtools)
<span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"HajkD/myTAI"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>, <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)

<span class="co"># On Windows, this won't work - see ?build_github_devtools</span>
<span class="co"># install_github("HajkD/myTAI", build_vignettes = TRUE, dependencies = TRUE)</span>

<span class="co"># When working with Windows, first you need to install the</span>
<span class="co"># R package: rtools </span>
<span class="co"># or consult: http://www.rstudio.com/products/rpackages/devtools/</span>

<span class="co"># Afterwards you can install devtools -&gt; install.packages("devtools")</span>
<span class="co"># and then you can run:</span>

devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"HajkD/myTAI"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>, <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)

<span class="co"># and then call it from the library</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(<span class="st">"myTAI"</span>, <span class="dt">lib.loc =</span> <span class="st">"C:/Program Files/R/R-3.1.1/library"</span>)</code></pre></div>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<p>Domazet-Lošo T. and Tautz D. <strong>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</strong>. <em>Nature</em> (2010) 468: 815-8.</p>
<p>Quint M, Drost HG, et al. <strong>A transcriptomic hourglass in plant embryogenesis</strong>. <em>Nature</em> (2012) 490: 98-101.</p>
<p>Drost HG, Gabel A, Grosse I, Quint M. <strong>Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis</strong>. <em>Mol. Biol. Evol.</em> (2015) 32 (5): 1221-1231.</p>
<p>Drost HG, Bellstädt J, Ó’Maoiléidigh DS, Silva AT, Gabel A, Weinholdt C, Ryan PT, Dekkers BJW, Bentsink L, Hilhorst H, Ligterink W, Wellmer F, Grosse I, and Quint M. <strong>Post-embryonic hourglass patterns mark ontogenetic transitions in plant development</strong>. <em>Mol. Biol. Evol.</em> (2016) <a href="http://mbe.oxfordjournals.org/content/early/2016/02/23/molbev.msw039.short?rss=1">doi:10.1093/molbev/msw039</a></p>
</div>
<div id="discussions-and-bug-reports" class="section level2">
<h2 class="hasAnchor">
<a href="#discussions-and-bug-reports" class="anchor"></a>Discussions and Bug Reports</h2>
<p>I would be very happy to learn more about potential improvements of the concepts and functions provided in this package.</p>
<p>Furthermore, in case you find some bugs or need additional (more flexible) functionality of parts of this package, please let me know:</p>
<p><a href="https://github.com/HajkD/myTAI/issues" class="uri">https://github.com/HajkD/myTAI/issues</a></p>
</div>
<div id="acknowledgement" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgement" class="anchor"></a>Acknowledgement</h2>
<p>I would like to thank several individuals for making this project possible.</p>
<p>First I would like to thank Ivo Grosse and Marcel Quint for providing me a place and the environment to be able to work on fascinating topics of Evo-Devo research and for the fruitful discussions that led to projects like this one.</p>
<p>Furthermore, I would like to thank Alexander Gabel and Jan Grau for valuable discussions on how to improve some methodological concepts of some analyses present in this package.</p>
<p>I would also like to thank Master Students: Sarah Scharfenberg, Anne Hoffmann, and Sebastian Wussow who worked intensively with this package and helped me to improve the usability and logic of the package environment.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=myTAI">https://​cloud.r-project.org/​package=myTAI</a>
</li>
<li>Browse source code at <br><a href="https://github.com/HajkD/myTAI">https://​github.com/​HajkD/​myTAI</a>
</li>
<li>Report a bug at <br><a href="https://github.com/HajkD/myTAI/issues">https://​github.com/​HajkD/​myTAI/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing myTAI</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Hajk-Georg Drost <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/&lt;https://orcid.org/0000-0002-1567-306X&gt;" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/HajkD/myTAI"><img src="https://travis-ci.org/HajkD/myTAI.svg?branch=master" alt="Travis-CI Build Status"></a></li>
<li><a href="https://www.rpackages.io/package/myTAI"><img src="https://www.rpackages.io/badge/myTAI.svg" alt="rpackages.io rank"></a></li>
<li><a href="https://github.com/metacran/cranlogs.app"><img src="http://cranlogs.r-pkg.org/badges/myTAI" alt="rstudio mirror downloads"></a></li>
<li><a href="https://github.com/metacran/cranlogs.app"><img src="http://cranlogs.r-pkg.org/badges/grand-total/myTAI" alt="rstudio mirror downloads"></a></li>
</ul>
</div>
</div>
</div>

      <footer><div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
