<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Perform Reductive Early Conservation Test — EarlyConservationTest • myTAI</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
<meta property="og:title" content="Perform Reductive Early Conservation Test — EarlyConservationTest" />

<meta property="og:description" content="The Reductive Early Conservation Test aims to statistically evaluate the
existence of a monotonically increasing phylotranscriptomic pattern based on TAI or TDI computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an early conservation like pattern. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an early conservation (low-high-high) shape." />
<meta name="twitter:card" content="summary" />


<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->




  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.7.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform Reductive Early Conservation Test</h1>
    <small>Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/EarlyConservationTest.R'><code>R/EarlyConservationTest.R</code></a></small>
    <div class="hidden name"><code>EarlyConservationTest.Rd</code></div>
    </div>

    
    <p>The <em>Reductive Early Conservation Test</em> aims to statistically evaluate the
existence of a monotonically increasing phylotranscriptomic pattern based on <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an early conservation like pattern. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an early conservation (low-high-high) shape.</p>
    

    <pre class="usage"><span class='fu'>EarlyConservationTest</span>(<span class='no'>ExpressionSet</span>, <span class='kw'>modules</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>permutations</span> <span class='kw'>=</span> <span class='fl'>1000</span>,
  <span class='kw'>lillie.test</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>plotHistogram</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>runs</span> <span class='kw'>=</span> <span class='fl'>10</span>,
  <span class='kw'>parallel</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>gof.warning</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>custom.perm.matrix</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>modules</th>
      <td><p>a list storing three elements: early, mid, and late. Each element expects a numeric
vector specifying the developmental stages or experiments that correspond to each module. 
For example, <code>module</code> = list(early = 1:2, mid = 3:5, late = 6:7) devides a dataset 
storing seven developmental stages into 3 modules.</p></td>
    </tr>
    <tr>
      <th>permutations</th>
      <td><p>a numeric value specifying the number of permutations to be performed for the <code>ReductiveHourglassTest</code>.</p></td>
    </tr>
    <tr>
      <th>lillie.test</th>
      <td><p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed to quantify the goodness of fit.</p></td>
    </tr>
    <tr>
      <th>plotHistogram</th>
      <td><p>a boolean value specifying whether a <em>Lillifor's Kolmogorov-Smirnov-Test</em> 
shall be performed to test the goodness of fit of the approximated distribution, as well as additional plots quantifying the significance
of the observed phylotranscriptomic pattern.</p></td>
    </tr>
    <tr>
      <th>runs</th>
      <td><p>specify the number of runs to be performed for goodness of fit computations, in case <code>plotHistogram</code> = <code>TRUE</code>.
In most cases <code>runs</code> = 100 is a reasonable choice. Default is <code>runs</code> = 10 (because it takes less computation time for demonstration purposes).</p></td>
    </tr>
    <tr>
      <th>parallel</th>
      <td><p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p></td>
    </tr>
    <tr>
      <th>gof.warning</th>
      <td><p>a logical value indicating whether non significant goodness of fit results should be printed as warning. Default is <code>gof.warning = FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>custom.perm.matrix</th>
      <td><p>a custom <code><a href='bootMatrix.html'>bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a list object containing the list elements:</p>
<p><code>p.value</code> : the p-value quantifying the statistical significance (low-high-high pattern) of the given phylotranscriptomics pattern.</p>
<p><code>std.dev</code> : the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.</p>
<p><code>lillie.test</code> : a boolean value specifying whether the <em>Lillifors KS-Test</em> returned a p-value &gt; 0.05, 
which indicates that fitting the permuted scores with a normal distribution seems plausible.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <em>reductive early conservation test</em> is a permutation test based on the following test statistic.</p>
<p>(1) A set of developmental stages is partitioned into three modules - early, mid, and late - based on prior biological knowledge.</p>
<p>(2) The mean <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> value for each of the three modules T_early, T_mid, and T_late are computed.</p>
<p>(3) The two differences D1 = T_mid - T_early and D2 = T_late - T_early are calculated.</p>
<p>(4) The minimum D_min of D1 and D2 is computed as final test statistic of the reductive hourglass test.</p>
    <p>In order to determine the statistical significance of an observed minimum difference D_min 
the following permutation test was performed. Based on the <code><a href='bootMatrix.html'>bootMatrix</a></code> D_min 
is calculated from each of the permuted <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> profiles, 
approximated by a Gaussian distribution with method of moments estimated parameters returned by <code><a href='http://www.rdocumentation.org/packages/fitdistrplus/topics/fitdist'>fitdist</a></code>, 
and the corresponding p-value is computed by <code>pnorm</code> given the estimated parameters of the Gaussian distribution. 
The <em>goodness of fit</em> for the random vector <em>D_min</em> is statistically quantified by an Lilliefors (Kolmogorov-Smirnov) test 
for normality.</p>
    <p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:</p>       
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href='http://www.rdocumentation.org/packages/fitdistrplus/topics/descdist'>descdist</a></code>. 
This plot illustrates which distributions seem plausible to fit the resulting permutation vector D_min. 
In the case of the <em>reductive early conservation test</em> a normal distribution seemed plausible.</p>
<p>(2) A histogram of D_min combined with the density plot is plotted. D_min is then fitted by a normal distribution. 
The corresponding parameters are estimated by <em>moment matching estimation</em> using the <code><a href='http://www.rdocumentation.org/packages/fitdistrplus/topics/fitdist'>fitdist</a></code> function.</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.</p>
<p>(4) A barplot showing the number of cases in which the underlying goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test 
for normality) has shown to be significant (<code>TRUE</code>) or not significant (<code>FALSE</code>). 
This allows to quantify the permutation bias and their implications on the goodness of fit.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Drost HG et al. (2015) <em>Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis</em>. Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012.</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.</p>
<p>Piasecka B, Lichocki P, Moretti S, et al. (2013) <em>The hourglass and the early conservation models co-existing
patterns of developmental constraints in vertebrates</em>. PLoS Genet. 9(4): e1003476.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='ecScore.html'>ecScore</a></code>, <code><a href='bootMatrix.html'>bootMatrix</a></code>, <code><a href='FlatLineTest.html'>FlatLineTest</a></code>,<code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code> , <code><a href='PlotPattern.html'>PlotPattern</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'>data</span>(<span class='no'>PhyloExpressionSetExample</span>)

<span class='co'># perform the early conservation test for a PhyloExpressionSet</span>
<span class='co'># here the prior biological knowledge is that stages 1-2 correspond to module 1 = early,</span>
<span class='co'># stages 3-5 to module 2 = mid (phylotypic module), and stages 6-7 correspond to</span>
<span class='co'># module 3 = late</span>
<span class='fu'>EarlyConservationTest</span>(<span class='no'>PhyloExpressionSetExample</span>,
                       <span class='kw'>modules</span> <span class='kw'>=</span> <span class='fu'>list</span>(<span class='kw'>early</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>2</span>, <span class='kw'>mid</span> <span class='kw'>=</span> <span class='fl'>3</span>:<span class='fl'>5</span>, <span class='kw'>late</span> <span class='kw'>=</span> <span class='fl'>6</span>:<span class='fl'>7</span>),
                       <span class='kw'>permutations</span> <span class='kw'>=</span> <span class='fl'>1000</span>)</div><div class='output co'>#&gt; $p.value
#&gt; [1] 0.9998301
#&gt; 
#&gt; $std.dev
#&gt; [1] 0.05555726 0.05363391 0.05197907 0.05070580 0.05007097 0.05193954 0.05605278
#&gt; 
#&gt; $lillie.test
#&gt; [1] NA
#&gt; </div><div class='input'>

<span class='co'># use your own permutation matrix based on which p-values (EarlyConservationTest)</span>
<span class='co'># shall be computed</span>
<span class='no'>custom_perm_matrix</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='bootMatrix.html'>bootMatrix</a></span>(<span class='no'>PhyloExpressionSetExample</span>,<span class='fl'>100</span>)

<span class='fu'>EarlyConservationTest</span>(<span class='no'>PhyloExpressionSetExample</span>,
                       <span class='kw'>modules</span> <span class='kw'>=</span> <span class='fu'>list</span>(<span class='kw'>early</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>2</span>, <span class='kw'>mid</span> <span class='kw'>=</span> <span class='fl'>3</span>:<span class='fl'>5</span>, <span class='kw'>late</span> <span class='kw'>=</span> <span class='fl'>6</span>:<span class='fl'>7</span>),
                       <span class='kw'>custom.perm.matrix</span> <span class='kw'>=</span> <span class='no'>custom_perm_matrix</span>)</div><div class='output co'>#&gt; $p.value
#&gt; [1] 0.9998574
#&gt; 
#&gt; $std.dev
#&gt;     Zygote   Quadrant   Globular      Heart    Torpedo       Bent     Mature 
#&gt; 0.05614129 0.05482230 0.05256984 0.05009957 0.04602865 0.04852778 0.04691451 
#&gt; 
#&gt; $lillie.test
#&gt; [1] NA
#&gt; </div><div class='input'>                       
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Hajk-Georg Drost

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

