<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Filter for Expressed Genes — Expressed • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Filter for Expressed Genes — Expressed" />

<meta property="og:description" content="This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined cut.off value. Hence, this function allows to remove
genes that have been defined as not expressed or outliers and returns an ExpressionSet retaining only expressed genes." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Filter for Expressed Genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/Expressed.R'><code>R/Expressed.R</code></a></small>
    <div class="hidden name"><code>Expressed.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined <code>cut.off</code> value. Hence, this function allows to remove
genes that have been defined as <em>not expressed</em> or <em>outliers</em> and returns an <code>ExpressionSet</code> retaining only expressed genes.</p>
    
    </div>

    <pre class="usage"><span class='fu'>Expressed</span>(<span class='no'>ExpressionSet</span>, <span class='no'>cut.off</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"const"</span>,
  <span class='kw'>comparison</span> <span class='kw'>=</span> <span class='st'>"below"</span>, <span class='kw'>n</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>cut.off</th>
      <td><p>a numeric value specifying the expression cut-off to define genes as <em>not expressed</em> (<code>comparison = "below"</code>) , <em>outliers</em> (<code>comparison = "above"</code>), or both (<code>comparison = "both"</code>). See <code>comparison</code> for details. In case <code>comparison = "both"</code>, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value at the second position.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>a method defining how to treat gene expression values in multiple stages. The corresponding method that is chosen allows to control the stage-wise fulfillment of the threshold criteria. Options are <code>"const"</code>, <code>"min-set"</code>, and <code>"n-set"</code>.</p></td>
    </tr>
    <tr>
      <th>comparison</th>
      <td><p>a character string specifying whether genes having expression levels
 below, above, or below AND above (both) the <code>cut.off</code> value should be excluded from the dataset.
 In case <code>comparison = "both"</code> is chosen, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value
at the second position.</p></td>
    </tr>
    <tr>
      <th>n</th>
      <td><p>a numeric value for <code>method = "n-set"</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This filter function allows users to remove genes from the <code>ExpressionSet</code> object that undercut or exceed a certain expression level <code>cut.off</code>.</p>
<p>Following extraction criteria are implemented in this function:</p><ul>
<li><p><code>const</code>: all genes that have at least one stage that undercuts or exceeds the expression <code>cut.off</code> will be excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage <code>ExpressionSet</code> genes passing the expression level <code>cut.off</code> in 6 stages will be retained in the <code>ExpressionSet</code>.</p></li>
<li><p><code>min-set</code>: genes passing the expression level <code>cut.off</code> in <code><a href='https://www.rdocumentation.org/packages/base/topics/Round'>ceiling(n/2)</a></code> stages will be retained in the <code>ExpressionSet</code>, where <em>n</em> is the number of stages in the <code>ExpressionSet</code>.</p></li>
<li><p><code>n-set</code>: genes passing the expression level <code>cut.off</code> in <code>n</code> stages will be retained in the <code>ExpressionSet</code>. Here, the argument <code>n</code> needs to be specified.</p></li>
</ul>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>PhyloExpressionSetExample</span>)

<span class='co'># remove genes that have an expression level below 8000 </span>
<span class='co'># in at least one developmental stage</span>
<span class='no'>FilterConst</span> <span class='kw'>&lt;-</span> <span class='fu'>Expressed</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                         <span class='kw'>cut.off</span>       <span class='kw'>=</span> <span class='fl'>8000</span>,
                         <span class='kw'>method</span>        <span class='kw'>=</span> <span class='st'>"const"</span>,
                         <span class='kw'>comparison</span>    <span class='kw'>=</span> <span class='st'>"below"</span>)

<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/dim'>dim</a></span>(<span class='no'>FilterConst</span>) <span class='co'># check number of retained genes</span></div><div class='output co'>#&gt; [1] 449   9</div><div class='input'>
<span class='co'># remove genes that have an expression level below 8000 </span>
<span class='co'># in at least 3 developmental stages </span>
<span class='co'># (in this case: ceiling(7/2) = 4 stages fulfilling the cut-off criteria)</span>
<span class='no'>FilterMinSet</span> <span class='kw'>&lt;-</span> <span class='fu'>Expressed</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                          <span class='kw'>cut.off</span>       <span class='kw'>=</span> <span class='fl'>8000</span>,
                          <span class='kw'>method</span>        <span class='kw'>=</span> <span class='st'>"min-set"</span>,
                          <span class='kw'>comparison</span>    <span class='kw'>=</span> <span class='st'>"below"</span>)

<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/dim'>dim</a></span>(<span class='no'>FilterMinSet</span>) <span class='co'># check number of retained genes</span></div><div class='output co'>#&gt; [1] 830   9</div><div class='input'>
<span class='co'># remove genes that have an expression level below 8000 </span>
<span class='co'># in at least 5 developmental stages (in this case: n = 2 stages fulfilling the criteria)</span>
<span class='no'>FilterNSet</span> <span class='kw'>&lt;-</span> <span class='fu'>Expressed</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                        <span class='kw'>cut.off</span>       <span class='kw'>=</span> <span class='fl'>8000</span>,
                        <span class='kw'>method</span>        <span class='kw'>=</span> <span class='st'>"n-set"</span>,
                        <span class='kw'>comparison</span>    <span class='kw'>=</span> <span class='st'>"below"</span>,
                        <span class='kw'>n</span>             <span class='kw'>=</span> <span class='fl'>2</span>)

<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/dim'>dim</a></span>(<span class='no'>FilterMinSet</span>) <span class='co'># check number of retained genes</span></div><div class='output co'>#&gt; [1] 830   9</div><div class='input'>


<span class='co'># remove expression levels that exceed the cut.off criteria</span>
<span class='no'>FilterMinSet</span> <span class='kw'>&lt;-</span> <span class='fu'>Expressed</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                          <span class='kw'>cut.off</span>       <span class='kw'>=</span> <span class='fl'>12000</span>,
                          <span class='kw'>method</span>        <span class='kw'>=</span> <span class='st'>"min-set"</span>,
                          <span class='kw'>comparison</span>    <span class='kw'>=</span> <span class='st'>"above"</span>)

<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/dim'>dim</a></span>(<span class='no'>FilterMinSet</span>) <span class='co'># check number of retained genes</span></div><div class='output co'>#&gt; [1] 24456     9</div><div class='input'>

<span class='co'># remove expression levels that undercut AND exceed the cut.off criteria</span>
<span class='no'>FilterMinSet</span> <span class='kw'>&lt;-</span> <span class='fu'>Expressed</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                          <span class='kw'>cut.off</span>       <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>8000</span>,<span class='fl'>12000</span>),
                          <span class='kw'>method</span>        <span class='kw'>=</span> <span class='st'>"min-set"</span>,
                          <span class='kw'>comparison</span>    <span class='kw'>=</span> <span class='st'>"both"</span>)

<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/dim'>dim</a></span>(<span class='no'>FilterMinSet</span>) <span class='co'># check number of retained genes</span></div><div class='output co'>#&gt; [1] 111   9</div><div class='input'>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

