<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix — MatchMap • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix — MatchMap" />

<meta property="og:description" content="This function matches a Phylostratigraphic Map or Divergence Map only storing unique gene ids with a ExpressionMatrix
also storing only unique gene ids." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.8.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/MatchMap.R'><code>R/MatchMap.R</code></a></small>
    <div class="hidden name"><code>MatchMap.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function matches a <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> only storing unique gene ids with a ExpressionMatrix
also storing only unique gene ids.</p>
    
    </div>

    <pre class="usage"><span class='fu'>MatchMap</span>(<span class='no'>Map</span>, <span class='no'>ExpressionMatrix</span>, <span class='kw'>remove.duplicates</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>accumulate</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>Map</th>
      <td><p>a standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> object.</p></td>
    </tr>
    <tr>
      <th>ExpressionMatrix</th>
      <td><p>a standard ExpressionMatrix object.</p></td>
    </tr>
    <tr>
      <th>remove.duplicates</th>
      <td><p>a logical value indicating whether duplicate gene ids should be removed from the data set.</p></td>
    </tr>
    <tr>
      <th>accumulate</th>
      <td><p>an accumulation function such as <code><a href='https://www.rdocumentation.org/packages/base/topics/mean'>mean()</a></code>, <code><a href='https://www.rdocumentation.org/packages/stats/topics/median'>median()</a></code>, or <code><a href='https://www.rdocumentation.org/packages/base/topics/Extremes'>min()</a></code>
to accumulate multiple expression levels that map to the same unique gene id present in the <code>ExpressionMatrix</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>In phylotranscriptomics analyses two major techniques are performed to
obtain standard <em>Phylostratigraphic map</em> or <em>Divergence map</em> objects.</p>
<p>To obtain a <em>Phylostratigraphic Map</em>, <em>Phylostratigraphy</em> (Domazet-Loso et al., 2007) has to be performed. To obtain a <em>Divergence Map</em>, 
orthologous gene detection, Ka/Ks computations, and decilation (Quint et al., 2012; Drost et al., 2015) have to be performed.</p>
<p>The resulting standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> objects consist of 2 colums storing the phylostratum assignment 
or divergence stratum assignment of a given gene in column one, and the corresponding gene id of that gene on columns two.</p>
<p>A standard ExpressionMatrix is a common gene expression matrix storing the gene ids or probe ids in the first column, and all experiments/stages/replicates in the following columns.</p>
<p>The <em>MatchMap</em> function takes both standard datasets: <em>Map</em> and <em>ExpressionMatrix</em> to merge both data sets to obtain a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
<p>This procedure is analogous to <code><a href='https://www.rdocumentation.org/packages/base/topics/merge'>merge</a></code>, but is customized to the <em>Phylostratigraphic Map</em>, <em>Divergence Map</em>, and <em>ExpressionMatrix</em> standards to allow a faster and more intuitive usage.</p>
<p>In case you work with an ExpressionMatrix that stores multiple expression levels for a unique gene id, you
can specify the <code>accumulation</code> argument to accumulate these multiple expression levels to obtain
one expression level for one unique gene.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Domazet-Loso T, Brajkovic J, Tautz D (2007) A phylostratigraphy approach to uncover the genomic history of major adaptations in metazoan lineages. Trends Genet. 23: 533-9.</p>
<p>Domazet-Loso T, Tautz D (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.</p>
<p>Drost HG et al. (2015) <em>Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis</em>. Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># load a standard PhyloExpressionSet</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>PhyloExpressionSetExample</span>)

<span class='co'># in a standard PhyloExpressionSet, </span>
<span class='co'># column one and column two denote a standard </span>
<span class='co'># phylostratigraphic map</span>
<span class='no'>PhyloMap</span> <span class='kw'>&lt;-</span> <span class='no'>PhyloExpressionSetExample</span>[ , <span class='fl'>1</span>:<span class='fl'>2</span>]

<span class='co'># look at the phylostratigraphic map standard</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='no'>PhyloMap</span>)</div><div class='output co'>#&gt;   Phylostratum      GeneID
#&gt; 1            1 at1g01040.2
#&gt; 2            1 at1g01050.1
#&gt; 3            1 at1g01070.1
#&gt; 4            1 at1g01080.2
#&gt; 5            1 at1g01090.1
#&gt; 6            1 at1g01120.1</div><div class='input'>
<span class='co'># in a standard PhyloExpressionSet, column two combined </span>
<span class='co'># with column 3 - N denote a standard ExpressionMatrix</span>
<span class='no'>ExpressionMatrixExample</span> <span class='kw'>&lt;-</span> <span class='no'>PhyloExpressionSetExample</span>[ , <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>2</span>,<span class='fl'>3</span>:<span class='fl'>9</span>)]

<span class='co'># these two data sets shall illustrate an example </span>
<span class='co'># phylostratigraphic map that is returned</span>
<span class='co'># by a standard phylostratigraphy run, and a expression set </span>
<span class='co'># that is the result of expression data analysis </span>
<span class='co'># (background correction, normalization, ...)</span>

<span class='co'># now we can use the MatchMap function to merge both data sets</span>
<span class='co'># to obtain a standard PhyloExpressionSet</span>

<span class='no'>PES</span> <span class='kw'>&lt;-</span> <span class='fu'>MatchMap</span>(<span class='no'>PhyloMap</span>, <span class='no'>ExpressionMatrixExample</span>)

<span class='co'># note that the function returns a head() </span>
<span class='co'># of the matched gene ids to enable</span>
<span class='co'># the user to find potential mis-matches</span>

<span class='co'># the entire procedure is analogous to merge() </span>
<span class='co'># with two data sets sharing the same gene ids </span>
<span class='co'># as column (primary key)</span>
<span class='no'>PES_merge</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/merge'>merge</a></span>(<span class='no'>PhyloMap</span>, <span class='no'>ExpressionMatrixExample</span>)</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.2.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

