<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot the Transcriptome Age Index or Transcriptome Divergence Index — PlotPattern • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Plot the Transcriptome Age Index or Transcriptome Divergence Index — PlotPattern" />

<meta property="og:description" content="Function to plot the TAI or TDI of a given PhyloExpressionSet or DivergenceExpressionSet object.
This function plot the TAI or TDI of a given PhyloExpressionSet or DivergenceExpressionSet object." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the Transcriptome Age Index or Transcriptome Divergence Index</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/PlotPattern.R'><code>R/PlotPattern.R</code></a></small>
    <div class="hidden name"><code>PlotPattern.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Function to plot the TAI or TDI of a given PhyloExpressionSet or DivergenceExpressionSet object.
This function plot the <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> of a given PhyloExpressionSet or DivergenceExpressionSet object.</p>
    
    </div>

    <pre class="usage"><span class='fu'>PlotPattern</span>(<span class='no'>ExpressionSet</span>, <span class='kw'>TestStatistic</span> <span class='kw'>=</span> <span class='st'>"FlatLineTest"</span>,
  <span class='kw'>modules</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>permutations</span> <span class='kw'>=</span> <span class='fl'>1000</span>, <span class='kw'>lillie.test</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>digits.ylab</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>p.value</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>shaded.area</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>y.ticks</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>custom.perm.matrix</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>TestStatistic</th>
      <td><p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Possible values can be: <code>TestStatistic</code> = <code>"FlatLineTest"</code> : Statistical test for the deviation from a flat line.
<code>TestStatistic</code> = <code>"ReductiveHourglassTest"</code> : Statistical test for the existence of a hourglass shape (high-low-high pattern).
<code>TestStatistic</code> = <code>"EarlyConservationTest"</code> : Statistical test for the existence of a earlyconservation pattern (low-high-high pattern).</p></td>
    </tr>
    <tr>
      <th>modules</th>
      <td><p>a list storing three elements for the <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code> or <code><a href='EarlyConservationTest.html'>EarlyConservationTest</a></code>: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example, 
<code>module</code> = <code><a href='https://www.rdocumentation.org/packages/base/topics/list'>list(early = 1:2, mid = 3:5, late = 6:7)</a></code> devides a dataset storing seven developmental stages into 3 modules.</p></td>
    </tr>
    <tr>
      <th>permutations</th>
      <td><p>a numeric value specifying the number of permutations to be performed for the <code><a href='FlatLineTest.html'>FlatLineTest</a></code>, <code><a href='EarlyConservationTest.html'>EarlyConservationTest</a></code> or <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code>.</p></td>
    </tr>
    <tr>
      <th>lillie.test</th>
      <td><p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed.</p></td>
    </tr>
    <tr>
      <th>digits.ylab</th>
      <td><p>a numeric value specifying the number of digits shown for the TAI or TDI values on the y-axis.</p></td>
    </tr>
    <tr>
      <th>p.value</th>
      <td><p>a boolean value specifying whether the p-value of the test statistic shall be printed within the plot area.</p></td>
    </tr>
    <tr>
      <th>shaded.area</th>
      <td><p>a boolean value specifying whether a shaded area shall 
be drawn for the developmental stages defined to be the presumptive phylotypic period.</p></td>
    </tr>
    <tr>
      <th>y.ticks</th>
      <td><p>a numeric value specifying the number of ticks to be drawn on the y-axis.</p></td>
    </tr>
    <tr>
      <th>custom.perm.matrix</th>
      <td><p>a custom <code><a href='bootMatrix.html'>bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic visualized by <code>PlotPattern</code>. Default is <code>custom.perm.matrix = NULL</code>.</p></td>
    </tr>
    <tr>
      <th>&#8230;</th>
      <td><p>default plot parameters.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a plot visualizing the phylotranscriptomic pattern of a given PhyloExpressionSet or DivergenceExpressionSet object.</p>
<p>The corresponding <em>p-value</em> of the test statistic is named as follows:</p>
<p><code>p_flt</code> = p-value of the corresponding <code><a href='FlatLineTest.html'>FlatLineTest</a></code></p>
<p><code>p_rht</code> = p-value of the corresponding <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code></p>
<p><code>p_ect</code> = p-value of the corresponding <code><a href='EarlyConservationTest.html'>EarlyConservationTest</a></code></p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function computes a permutation test quantifying the statistical significance of the prensent phylotranscriptomics pattern. 
The user can choose between the <code><a href='FlatLineTest.html'>FlatLineTest</a></code>, <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code>, or <code><a href='EarlyConservationTest.html'>EarlyConservationTest</a></code>. 
The <code><a href='FlatLineTest.html'>FlatLineTest</a></code> tests for any significant deviation from a flat line. 
Each period or stage that significantly deviates from a flat line, might be governed by stronger selective pressure (in terms of natural selection) compared to other stages or periods of development.
The <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code> specificly tests for the statistical significance of a molecular hourglass pattern (high-low-high pattern) with prior biological knowlegde.
The corresponding p-value that is printed within the plot (by default) specifies the statistical significance of the chosen test statistic.</p>
<p>The <code><a href='EarlyConservationTest.html'>EarlyConservationTest</a></code> specificly tests for the statistical significance of a low-high-high pattern (monotonically increasing function)
with prior biological knowlegde.
The corresponding p-value that is printed within the plot (by default) specifies the statistical significance of the chosen test statistic.</p>
<p>The x-axis denotes the developmental series (time course / experiments / ontogeny) of the input ExpressionSet and the y-axis 
denotes the corresponding mean transcriptome age value (<code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code>) of the given ExpressionSet.</p>
<p>Furthermore, the grey lines above and below the actual phylotranscriptomics pattern denotes the standard deviations of <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> 
values that have been estimated from the <code><a href='bootMatrix.html'>bootMatrix</a></code>.
A low mean transcriptome age value denotes an evolutionary older transcriptome being active during the corresponding periods, 
whereas a high mean transcriptome age value denotes an evolutionary younger transcriptome being active during the corresponding periods.
For mean transcriptome divergence, a low mean transcriptome divergence value denotes a more conserved transcriptome 
being active (between two species), whereas a high mean transcriptome divergence value denotes a more divergent transcriptome 
being active (between two species) - in terms of protein-sequence substitution rates.</p>
<p>This function is useful to fastly plot the <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> 
profile of a given PhyloExpressionSet or DivergenceExpressionSet object and 
the statistical significance of the corresponding pattern.
Internally the function calls several graphics functions, 
such as <code><a href='https://www.rdocumentation.org/packages/graphics/topics/plot'>plot</a></code>, <code><a href='https://www.rdocumentation.org/packages/graphics/topics/axis'>axis</a></code>, and <code><a href='https://www.rdocumentation.org/packages/graphics/topics/legend'>legend</a></code>. 
For the ellipsis argument <code><a href='https://www.rdocumentation.org/packages/base/topics/dots'>...</a></code> all graphics specific arguments can be defined. 
Internally the function specific arguments for e.g. <code><a href='https://www.rdocumentation.org/packages/graphics/topics/plot'>plot</a></code>, <code><a href='https://www.rdocumentation.org/packages/graphics/topics/axis'>axis</a></code>, 
and <code><a href='https://www.rdocumentation.org/packages/graphics/topics/legend'>legend</a></code> will be detected and are passed to the corresponding function.</p>
<p>Hence, when calling the function <code>PlotPattern</code>, one can specify arguments 
for <code><a href='https://www.rdocumentation.org/packages/graphics/topics/plot'>plot</a></code> and <code><a href='https://www.rdocumentation.org/packages/graphics/topics/axis'>axis</a></code> and <code><a href='https://www.rdocumentation.org/packages/graphics/topics/legend'>legend</a></code> 
as <code><a href='https://www.rdocumentation.org/packages/base/topics/dots'>...</a></code>.</p>
<p>In case prior biological knowledge is present for a specific period of development,
the <code>shaded.area</code> argument can be set to <code>TRUE</code> and the function will use
the values stored in the <code>mid</code> argument to draw a shaded area within the corresponding period of development.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Domazet-Loso T and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.</p>
<p>Drost HG et al. (2015) <em>Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis</em>. Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='TAI.html'>TAI</a></code>, <code><a href='TDI.html'>TDI</a></code>, <code><a href='FlatLineTest.html'>FlatLineTest</a></code>,
 <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code>, <code><a href='EarlyConservationTest.html'>EarlyConservationTest</a></code>,
 <code><a href='PlotSignature.html'>PlotSignature</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># load PhyloExpressionSet</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>PhyloExpressionSetExample</span>)

<span class='co'># only visualize the TAI profile without any test statistics...</span>
<span class='co'># this is equavalent to performing: plot(TAI(PhyloExpressionSetExample), type = "l", lwd = 6)</span>
<span class='fu'>PlotPattern</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
            <span class='kw'>TestStatistic</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
            <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"l"</span>,
            <span class='kw'>xlab</span>          <span class='kw'>=</span> <span class='st'>"Ontogeny"</span>,
            <span class='kw'>ylab</span>          <span class='kw'>=</span> <span class='st'>"TAI"</span>,
            <span class='kw'>lwd</span>           <span class='kw'>=</span> <span class='fl'>9</span>)</div><div class='img'><img src='PlotPattern-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'># the simplest example of plotting the TAI profile of a given PhyloExpressionSet:</span>
<span class='co'># In this case (default) the FlatLineTest will be performed to quantify</span>
<span class='co'># the statistical significance of the present TAI pattern and will be drawn as 'p = ... '</span>
<span class='co'># in the plot</span>

<span class='fu'>PlotPattern</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
            <span class='kw'>TestStatistic</span> <span class='kw'>=</span> <span class='st'>"FlatLineTest"</span>,
            <span class='kw'>permutations</span>  <span class='kw'>=</span> <span class='fl'>100</span>,
            <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"l"</span>,
            <span class='kw'>xlab</span>          <span class='kw'>=</span> <span class='st'>"Ontogeny"</span>,
            <span class='kw'>ylab</span>          <span class='kw'>=</span> <span class='st'>"TAI"</span>,
            <span class='kw'>lwd</span>           <span class='kw'>=</span> <span class='fl'>9</span>)</div><div class='img'><img src='PlotPattern-2.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'># an example performing the ReductiveHourglassTest and printing the p-value</span>
<span class='co'># and shaded area of the presumptive phylotypic period into the plot</span>
<span class='co'># Here the 'p = ...' denotes the p-value that is returned by the ReductiveHourglassTest</span>

<span class='fu'>PlotPattern</span>(
            <span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
            <span class='kw'>TestStatistic</span> <span class='kw'>=</span> <span class='st'>"ReductiveHourglassTest"</span>,
            <span class='kw'>modules</span>       <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='kw'>early</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>2</span>,<span class='kw'>mid</span> <span class='kw'>=</span> <span class='fl'>3</span>:<span class='fl'>5</span>,<span class='kw'>late</span> <span class='kw'>=</span> <span class='fl'>6</span>:<span class='fl'>7</span>),
            <span class='kw'>permutations</span>  <span class='kw'>=</span> <span class='fl'>100</span>,
            <span class='kw'>p.value</span>       <span class='kw'>=</span> <span class='fl'>TRUE</span>,
            <span class='kw'>shaded.area</span>   <span class='kw'>=</span> <span class='fl'>TRUE</span>,
            <span class='kw'>xlab</span>          <span class='kw'>=</span> <span class='st'>"Ontogeny"</span>,
            <span class='kw'>ylab</span>          <span class='kw'>=</span> <span class='st'>"TAI"</span>,
            <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"l"</span>,
            <span class='kw'>lwd</span>           <span class='kw'>=</span> <span class='fl'>9</span>)</div><div class='img'><img src='PlotPattern-3.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'># testing for early conservation model </span>
<span class='fu'>PlotPattern</span>( <span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
             <span class='kw'>TestStatistic</span> <span class='kw'>=</span> <span class='st'>"EarlyConservationTest"</span>,
            <span class='kw'>modules</span>        <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='kw'>early</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>2</span>,<span class='kw'>mid</span> <span class='kw'>=</span> <span class='fl'>3</span>:<span class='fl'>5</span>,<span class='kw'>late</span> <span class='kw'>=</span> <span class='fl'>6</span>:<span class='fl'>7</span>),
            <span class='kw'>permutations</span>   <span class='kw'>=</span> <span class='fl'>100</span>,
            <span class='kw'>p.value</span>        <span class='kw'>=</span> <span class='fl'>TRUE</span>,
            <span class='kw'>shaded.area</span>    <span class='kw'>=</span> <span class='fl'>TRUE</span>,
            <span class='kw'>xlab</span>           <span class='kw'>=</span> <span class='st'>"Ontogeny"</span>,
            <span class='kw'>ylab</span>           <span class='kw'>=</span> <span class='st'>"TAI"</span>,
            <span class='kw'>type</span>           <span class='kw'>=</span> <span class='st'>"l"</span>,
            <span class='kw'>lwd</span>            <span class='kw'>=</span> <span class='fl'>9</span>)</div><div class='img'><img src='PlotPattern-4.png' alt='' width='700' height='433' /></div><div class='input'>

<span class='co'># use your own permutation matrix</span>
<span class='no'>custom_perm_matrix</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='bootMatrix.html'>bootMatrix</a></span>(<span class='no'>PhyloExpressionSetExample</span>,<span class='fl'>100</span>)

<span class='fu'>PlotPattern</span>(<span class='kw'>ExpressionSet</span>      <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
            <span class='kw'>TestStatistic</span>      <span class='kw'>=</span> <span class='st'>"FlatLineTest"</span>,
            <span class='kw'>custom.perm.matrix</span> <span class='kw'>=</span> <span class='no'>custom_perm_matrix</span>,
            <span class='kw'>type</span>               <span class='kw'>=</span> <span class='st'>"l"</span>,
            <span class='kw'>xlab</span>               <span class='kw'>=</span> <span class='st'>"Ontogeny"</span>,
            <span class='kw'>ylab</span>               <span class='kw'>=</span> <span class='st'>"TAI"</span>,
            <span class='kw'>lwd</span>                <span class='kw'>=</span> <span class='fl'>9</span>)</div><div class='img'><img src='PlotPattern-5.png' alt='' width='700' height='433' /></div><div class='input'>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.2.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

