<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot the Phylostratum or Divergence Stratum Enrichment of a given Gene Set — PlotEnrichment • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Plot the Phylostratum or Divergence Stratum Enrichment of a given Gene Set — PlotEnrichment" />

<meta property="og:description" content="This function computes and visualizes the significance of enriched (over or underrepresented) Phylostrata or Divergence Strata within an input test.set." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the Phylostratum or Divergence Stratum Enrichment of a given Gene Set</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/PlotEnrichment.R'><code>R/PlotEnrichment.R</code></a></small>
    <div class="hidden name"><code>PlotEnrichment.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function computes and visualizes the significance of enriched (over or underrepresented) Phylostrata or Divergence Strata within an input <code>test.set</code>.</p>
    
    </div>

    <pre class="usage"><span class='fu'>PlotEnrichment</span>(<span class='no'>ExpressionSet</span>, <span class='no'>test.set</span>, <span class='kw'>use.only.map</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>measure</span> <span class='kw'>=</span> <span class='st'>"log-foldchange"</span>, <span class='kw'>complete.bg</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>legendName</span> <span class='kw'>=</span> <span class='st'>""</span>,
  <span class='kw'>over.col</span> <span class='kw'>=</span> <span class='st'>"steelblue"</span>, <span class='kw'>under.col</span> <span class='kw'>=</span> <span class='st'>"midnightblue"</span>,
  <span class='kw'>epsilon</span> <span class='kw'>=</span> <span class='fl'>1e-05</span>, <span class='kw'>cex.legend</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>cex.asterisk</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>plot.bars</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>p.adjust.method</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object (in case <code>only.map = FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>test.set</th>
      <td><p>a character vector storing the gene ids for which PS/DS enrichment analyses should be performed.</p></td>
    </tr>
    <tr>
      <th>use.only.map</th>
      <td><p>a logical value indicating whether instead of a standard <code>ExpressionSet</code> only a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to this function.</p></td>
    </tr>
    <tr>
      <th>measure</th>
      <td><p>a character string specifying the measure that should be used to quantify over and under representation of PS/DS. Measures can either be <code>measure = "foldchange"</code> (odds) or <code>measure = "log-foldchange"</code> (log-odds).</p></td>
    </tr>
    <tr>
      <th>complete.bg</th>
      <td><p>a logical value indicating whether the entire background set
of the input <code>ExpressionSet</code> should be considered when performing Fisher's exact
test (<code>complete.bg = TRUE</code>) or whether genes that are stored in <code>test.set</code>
should be excluded from the background set before performing Fisher's exact test
(<code>complete.bg = FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>legendName</th>
      <td><p>a character string specifying whether "PS" or "DS" are used to compute relative expression profiles.</p></td>
    </tr>
    <tr>
      <th>over.col</th>
      <td><p>color of the overrepresentation bars.</p></td>
    </tr>
    <tr>
      <th>under.col</th>
      <td><p>color of the underrepresentation bars.</p></td>
    </tr>
    <tr>
      <th>epsilon</th>
      <td><p>a small value to shift values by epsilon to avoid log(0) computations.</p></td>
    </tr>
    <tr>
      <th>cex.legend</th>
      <td><p>the <code>cex</code> value for the legend.</p></td>
    </tr>
    <tr>
      <th>cex.asterisk</th>
      <td><p>the <code>cex</code> value for the asterisk.</p></td>
    </tr>
    <tr>
      <th>plot.bars</th>
      <td><p>a logical value specifying whether or not bars should be visualized or whether only <code>p.values</code> and <code>enrichment.matrix</code> should be returned.</p></td>
    </tr>
    <tr>
      <th>p.adjust.method</th>
      <td><p>correction method to adjust p-values for multiple comparisons (see <code><a href='https://www.rdocumentation.org/packages/stats/topics/p.adjust'>p.adjust</a></code> for possible methods). E.g., <code>p.adjust.method = "BH"</code> (Benjamini &amp; Hochberg (1995)) or <code>p.adjust.method = "bonferroni"</code> (Bonferroni correction).</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>default graphics parameters.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This <em>Phylostratum</em> or <em>Divergence Stratum</em> enrichment analysis is motivated 
by Sestak and Domazet-Loso (2015) who perform <em>Phylostratum</em> or <em>Divergence Stratum</em> enrichment analyses to correlate organ evolution with the origin of organ specific genes.</p>
<p>In detail this function takes the <em>Phylostratum</em> or <em>Divergence Stratum</em> distribution of all genes stored in the input <code>ExpressionSet</code> as background set and
the <em>Phylostratum</em> or <em>Divergence Stratum</em> distribution of the <code>test.set</code> and performes a <code><a href='https://www.rdocumentation.org/packages/stats/topics/fisher.test'>fisher.test</a></code> for each <em>Phylostratum</em> or <em>Divergence Stratum</em> to quantify the statistical significance of over- or underrepresentated <em>Phylostrata</em> or <em>Divergence Strata</em> within the set of selected <code>test.set</code> genes.</p>
<p>To visualize the odds or log-odds of over or underrepresented genes within the <code>test.set</code> the following procedure is performed:</p>
<ul>
<li><p>N_ij denotes the number of genes in group j and deriving from PS i, with <em>i = 1, .. , n</em> and where <em>j = 1</em> denotes the background set and <em>j = 2</em> denotes the <code>test.set</code></p></li>
<li><p>N_i. denotes the total number of genes within PS i</p></li>
<li><p>N_.j denotes the total number of genes within group j</p></li>
<li><p>N_.. is the total number of genes within all groups j and all PS i</p></li>
<li><p>f_ij = N_ij / N_.. and g_ij = f_ij / f_.j denote relative frequencies between groups</p></li>
<li><p>f_i. denotes the between group sum of f_ij</p></li>
</ul>
    <p>The result is the fold-change value (odds) denoted as C = g_i2 / f_i. which is visualized above and below zero.</p> 
<p>In case a large number of Phylostrata or Divergence Strata is included in the input 
 <code>ExpressionSet</code>, p-values returned by <code>PlotEnrichment</code> should be adjusted for
 multiple comparisons which can be done by specifying the <code>p.adjust.method</code> argument.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Sestak and Domazet-Loso (2015). Phylostratigraphic Profiles in Zebrafish Uncover Chordate Origins of the Vertebrate Brain. Mol. Biol. Evol. 32(2): 299-312.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='EnrichmentTest.html'>EnrichmentTest</a></code>, <code><a href='https://www.rdocumentation.org/packages/stats/topics/fisher.test'>fisher.test</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>PhyloExpressionSetExample</span>)

<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/Random'>set.seed</a></span>(<span class='fl'>123</span>)
<span class='no'>test_set</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/sample'>sample</a></span>(<span class='no'>PhyloExpressionSetExample</span>[ , <span class='fl'>2</span>],<span class='fl'>10000</span>)

<span class='co'>## Examples with complete.bg = TRUE</span>
<span class='co'>## Hence: the entire background set of the input ExpressionSet is considered </span>
<span class='co'>## when performing Fisher's exact test </span>

<span class='co'># measure: log-foldchange</span>
<span class='fu'>PlotEnrichment</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
               <span class='kw'>test.set</span>      <span class='kw'>=</span> <span class='no'>test_set</span> ,
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>measure</span>       <span class='kw'>=</span> <span class='st'>"log-foldchange"</span>)</div><div class='img'><img src='PlotEnrichment-1.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; $p.values
#&gt;        PS1        PS2        PS3        PS4        PS5        PS6        PS7 
#&gt; 0.68771282 0.45306323 0.08095321 0.51414436 0.97148218 0.27478755 0.03204539 
#&gt;        PS8        PS9       PS10       PS11       PS12 
#&gt; 0.71071584 0.41062985 0.33163365 0.82260340 0.44749411 
#&gt; 
#&gt; $enrichment.matrix
#&gt;            BG_Set     Test_Set
#&gt; PS1  -0.004013721  0.006103497
#&gt; PS2  -0.009421917  0.014260287
#&gt; PS3   0.070249042 -0.114260412
#&gt; PS4  -0.014434359  0.021752197
#&gt; PS5   0.002585053 -0.003953739
#&gt; PS6   0.031468144 -0.049383568
#&gt; PS7   0.142016032 -0.247865603
#&gt; PS8   0.027940831 -0.043708747
#&gt; PS9  -0.052813240  0.077041531
#&gt; PS10  0.036470542 -0.057494017
#&gt; PS11 -0.020107047  0.030153208
#&gt; PS12 -0.021160867  0.031704875
#&gt; </div><div class='input'>

<span class='co'># measure: foldchange</span>
<span class='fu'>PlotEnrichment</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
               <span class='kw'>test.set</span>      <span class='kw'>=</span> <span class='no'>test_set</span> ,
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>measure</span>       <span class='kw'>=</span> <span class='st'>"foldchange"</span>)</div><div class='img'><img src='PlotEnrichment-2.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; $p.values
#&gt;        PS1        PS2        PS3        PS4        PS5        PS6        PS7 
#&gt; 0.68771282 0.45306323 0.08095321 0.51414436 0.97148218 0.27478755 0.03204539 
#&gt;        PS8        PS9       PS10       PS11       PS12 
#&gt; 0.71071584 0.41062985 0.33163365 0.82260340 0.44749411 
#&gt; 
#&gt; $enrichment.matrix
#&gt;         BG_Set  Test_Set
#&gt; PS1  0.9972217  1.004240
#&gt; PS2  0.9934903  1.009934
#&gt; PS3  1.0499140 -1.082449
#&gt; PS4  0.9900438  1.015193
#&gt; PS5  1.0017940 -1.002745
#&gt; PS6  1.0220552 -1.034828
#&gt; PS7  1.1035387 -1.187648
#&gt; PS8  1.0195738 -1.030789
#&gt; PS9  0.9640288  1.054892
#&gt; PS10 1.0256084 -1.040668
#&gt; PS11 0.9861409  1.021149
#&gt; PS12 0.9854374  1.022223
#&gt; </div><div class='input'>

<span class='co'>## Examples with complete.bg = FALSE</span>
<span class='co'>## Hence: the test.set genes are excluded from the background set before</span>
<span class='co'>## Fisher's exact test is performed</span>


<span class='co'># measure: log-foldchange</span>
<span class='fu'>PlotEnrichment</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
               <span class='kw'>test.set</span>      <span class='kw'>=</span> <span class='no'>test_set</span> ,
                <span class='kw'>complete.bg</span>  <span class='kw'>=</span> <span class='fl'>FALSE</span>,
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>measure</span>       <span class='kw'>=</span> <span class='st'>"log-foldchange"</span>)</div><div class='img'><img src='PlotEnrichment-3.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; $p.values
#&gt;       PS1       PS2       PS3       PS4       PS5       PS6       PS7       PS8 
#&gt; 0.7866977 0.6270033 0.2553465 0.6573481 1.0000000 0.4761380 0.1507753 0.8196041 
#&gt;       PS9      PS10      PS11      PS12 
#&gt; 0.5824149 0.5164466 0.8910157 0.6100468 
#&gt; 
#&gt; $enrichment.matrix
#&gt;            BG_Set     Test_Set
#&gt; PS1  -0.001733622  0.004369875
#&gt; PS2  -0.004058664  0.010201623
#&gt; PS3   0.031496415 -0.082763996
#&gt; PS4  -0.006202476  0.015549721
#&gt; PS5   0.001120210 -0.002833530
#&gt; PS6   0.013834678 -0.035548890
#&gt; PS7   0.066079768 -0.181785836
#&gt; PS8   0.012262194 -0.031446554
#&gt; PS9  -0.022270692  0.054770838
#&gt; PS10  0.016074318 -0.041419699
#&gt; PS11 -0.008615890  0.021537317
#&gt; PS12 -0.009062745  0.022642131
#&gt; </div><div class='input'>

<span class='co'># measure: foldchange</span>
<span class='fu'>PlotEnrichment</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
               <span class='kw'>test.set</span>      <span class='kw'>=</span> <span class='no'>test_set</span> ,
               <span class='kw'>complete.bg</span>   <span class='kw'>=</span> <span class='fl'>FALSE</span>,
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>measure</span>       <span class='kw'>=</span> <span class='st'>"foldchange"</span>)</div><div class='img'><img src='PlotEnrichment-4.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; $p.values
#&gt;       PS1       PS2       PS3       PS4       PS5       PS6       PS7       PS8 
#&gt; 0.7866977 0.6270033 0.2553465 0.6573481 1.0000000 0.4761380 0.1507753 0.8196041 
#&gt;       PS9      PS10      PS11      PS12 
#&gt; 0.5824149 0.5164466 0.8910157 0.6100468 
#&gt; 
#&gt; $enrichment.matrix
#&gt;         BG_Set  Test_Set
#&gt; PS1  0.9987990  1.003034
#&gt; PS2  0.9971906  1.007097
#&gt; PS3  1.0220790 -1.059066
#&gt; PS4  0.9957096  1.010838
#&gt; PS5  1.0007770 -1.001967
#&gt; PS6  1.0096372 -1.024951
#&gt; PS7  1.0469121 -1.134430
#&gt; PS8  1.0085436 -1.022057
#&gt; PS9  0.9846708  1.038721
#&gt; PS10 1.0112071 -1.029134
#&gt; PS11 0.9940378  1.015061
#&gt; PS12 0.9937370  1.015820
#&gt; </div><div class='input'>               
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.2.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

