<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot the Expression Levels of each Age or Divergence Category as Boxplot, Violinplot, or Dotplot — PlotCategoryExpr • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Plot the Expression Levels of each Age or Divergence Category as Boxplot, Violinplot, or Dotplot — PlotCategoryExpr" />

<meta property="og:description" content="This function visualizes the expression level distribution of each phylostratum during each time point or experiment
as boxplot, dot plot, or violin plot enabling users to quantify the age (PS) or divergence (DS) category specific contribution to the
corresponding transcriptome." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the Expression Levels of each Age or Divergence Category as Boxplot, Violinplot, or Dotplot</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/PlotCategoryExpr.R'><code>R/PlotCategoryExpr.R</code></a></small>
    <div class="hidden name"><code>PlotCategoryExpr.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function visualizes the expression level distribution of each phylostratum during each time point or experiment
as boxplot, dot plot, or violin plot enabling users to quantify the age (PS) or divergence (DS) category specific contribution to the
corresponding transcriptome.</p>
    
    </div>

    <pre class="usage"><span class='fu'>PlotCategoryExpr</span>(<span class='no'>ExpressionSet</span>, <span class='no'>legendName</span>, <span class='kw'>test.stat</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>"category-centered"</span>, <span class='kw'>distr.type</span> <span class='kw'>=</span> <span class='st'>"boxplot"</span>,
  <span class='kw'>log.expr</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>gene.set</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>legendName</th>
      <td><p>a character string specifying whether "PS" or "DS" are used to compute relative expression profiles.</p></td>
    </tr>
    <tr>
      <th>test.stat</th>
      <td><p>a logical value indicating whether a Benjamini-Hochberg adjusted <code><a href='https://www.rdocumentation.org/packages/stats/topics/kruskal.test'>kruskal.test</a></code> should be applied to determine
significant differences in age or divergence category specific expression.</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>type of age or divergence category comparison. Specifications can be <code>type = "category-centered"</code> or <code>type = "stage-centered"</code>.</p></td>
    </tr>
    <tr>
      <th>distr.type</th>
      <td><p>format of visualizing age or divergence category specific expression distributions. Either <code>distr.type = "boxplot"</code>, <code>distr.type = "dotplot"</code>, or
<code>distr.type = "violin"</code>.</p></td>
    </tr>
    <tr>
      <th>log.expr</th>
      <td><p>a logical value specifying whether or not expression levels should internally be log2-transformed before visualization.</p></td>
    </tr>
    <tr>
      <th>gene.set</th>
      <td><p>a character vector storing the gene ids for which gene expression levels shall be visualized.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A boxplot, violin plot, or dot plot visualizing the gene expression levels of
different PS or DS categories.</p>
<p>Furthermore, the statistical test results returned from the <code><a href='https://www.rdocumentation.org/packages/stats/topics/kruskal.test'>kruskal.test</a></code> are printed
to the console.</p>
<p>(1) '*'   = P-Value &lt;= 0.05</p>
<p>(2) '**'  = P-Value &lt;= 0.005</p>
<p>(3) '***' = P-Value &lt;= 0.0005</p>
<p>(4) 'n.s.' = not significant = P-Value &gt; 0.05</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This way of visualizing the gene expression distribution of each age (PS) or divergence (DS) category during
all developmental stages or experiments allows users to detect specific age or divergence categories contributing significant
levels of gene expression to the underlying biological process (transcriptome).</p>
<p>This quantification allows users to conclude that genes originating in specific PS or DS contribute significantly more to the overall transcriptome
than other genes originating from different PS or DS categories. More specialized analyses such as <code><a href='PlotMeans.html'>PlotMeans</a></code>, <code><a href='PlotRE.html'>PlotRE</a></code>,
<code><a href='PlotBarRE.html'>PlotBarRE</a></code>, etc. will then allow to study the exact mean expression patterns of these age or divergence categories.</p>
<p>The statistical quantification of differences between expression levels of different age or divergence categories
is done by performing a <code><a href='https://www.rdocumentation.org/packages/stats/topics/kruskal.test'>kruskal.test</a></code> with Benjamini &amp; Hochberg p-value adjustment for multiple comparisons.</p>
<ul>
<li><p><code>type = "category-centered"</code> Here, the <code><a href='https://www.rdocumentation.org/packages/stats/topics/kruskal.test'>kruskal.test</a></code> quantifies the differences of gene expression between all combinations of age or divergence categories for each stage or experiment separately. Here, a significant p-value quantifies that there is at least one pairwise comparison for which age or divergence categories significantly differ in their gene expression distribution. This type of analysis allows users to detect stages or experiments that show high diviation between age or divergence category contributions to the overall transcriptome or no significant deviations of age or divergence categories, suggesting equal age or divergence category contributions to the overall transcriptome.</p></li>
<li><p><code>type = "stage-centered"</code> Here, the <code><a href='https://www.rdocumentation.org/packages/stats/topics/kruskal.test'>kruskal.test</a></code> quantifies the differences of gene expression between all stages or experiments for each age or divergence category separately. Hence, the test quantifies whether or not the gene expression distribution of a single age or divergence category significantly changes throughout development or experiments. This type of analysis allows users to detect specific age or divergence categories that significantly change their expression levels throughout development or experiments.</p></li>
</ul>
    <p>Argument Specifications:</p>
<p>Argument: type</p>
<ul>
<li><p><code>type = "category-centered"</code> This specification allows users to compare the differences between all age or
 divergence categories during the same stage or experiment.</p></li>
<li><p><code>type = "stage-centered"</code> This specification allows users to compare the differences between all age or
 divergence categories between stages or experiments.</p></li>
</ul>
    <p>Argument: distr.type</p>
<ul>
<li><p><code>distr.type = "boxplot"</code> This specification allows users to visualize the expression distribution of all PS or DS as boxplot.</p></li>
<li><p><code>distr.type = "violin"</code> This specification allows users to visualize the expression distribution of all PS or DS as violin plot.</p></li>
<li><p><code>distr.type = "dotplot"</code> This specification allows users to visualize the expression distribution of all PS or DS as dot plot.</p></li>
</ul>
    <p>Finally, users can specify a <code>gene.set</code> (a subset of genes included in the input <code>ExpressioSet</code>)
for which expression levels should be visualized as boxplot, dotplot, or violinplot.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='PlotMeans.html'>PlotMeans</a></code>, <code><a href='PlotRE.html'>PlotRE</a></code>, <code><a href='PlotBarRE.html'>PlotBarRE</a></code>, <code><a href='age.apply.html'>age.apply</a></code>,
<code><a href='pTAI.html'>pTAI</a></code>, <code><a href='pTDI.html'>pTDI</a></code>, <code><a href='pStrata.html'>pStrata</a></code>, <code><a href='pMatrix.html'>pMatrix</a></code>, <code><a href='TAI.html'>TAI</a></code>, <code><a href='TDI.html'>TDI</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>PhyloExpressionSetExample</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>DivergenceExpressionSetExample</span>)</div><span class='co'># NOT RUN {</span>
<span class='co'># category-centered visualization of PS specific expression level distributions (log-scale)</span>
<span class='fu'>PlotCategoryExpr</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                     <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
                     <span class='kw'>test.stat</span>     <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                     <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"category-centered"</span>,
                     <span class='kw'>distr.type</span>    <span class='kw'>=</span> <span class='st'>"boxplot"</span>,
                     <span class='kw'>log.expr</span>      <span class='kw'>=</span> <span class='fl'>TRUE</span>)


<span class='co'># stage-centered visualization of PS specific expression level distributions (log-scale)</span>
<span class='fu'>PlotCategoryExpr</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                     <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
                     <span class='kw'>test.stat</span>     <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                     <span class='kw'>distr.type</span>    <span class='kw'>=</span> <span class='st'>"boxplot"</span>,
                     <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"stage-centered"</span>,
                     <span class='kw'>log.expr</span>      <span class='kw'>=</span> <span class='fl'>TRUE</span>)



<span class='co'># category-centered visualization of PS specific expression level distributions (log-scale)</span>
<span class='co'># as violoin plot</span>
<span class='fu'>PlotCategoryExpr</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                     <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
                     <span class='kw'>test.stat</span>     <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                     <span class='kw'>distr.type</span>    <span class='kw'>=</span> <span class='st'>"violin"</span>,
                     <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"stage-centered"</span>,
                     <span class='kw'>log.expr</span>      <span class='kw'>=</span> <span class='fl'>TRUE</span>)




<span class='co'># analogous for DivergenceExpressionSets</span>
<span class='fu'>PlotCategoryExpr</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>DivergenceExpressionSetExample</span>,
                     <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"DS"</span>,
                     <span class='kw'>test.stat</span>     <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                     <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"category-centered"</span>,
                     <span class='kw'>distr.type</span>    <span class='kw'>=</span> <span class='st'>"boxplot"</span>,
                     <span class='kw'>log.expr</span>      <span class='kw'>=</span> <span class='fl'>TRUE</span>)


<span class='co'># visualize the expression levels of 500 example genes</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/Random'>set.seed</a></span>(<span class='fl'>234</span>)
<span class='no'>example.gene.set</span> <span class='kw'>&lt;-</span> <span class='no'>PhyloExpressionSetExample</span>[<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/sample'>sample</a></span>(<span class='fl'>1</span>:<span class='fl'>25260</span>,<span class='fl'>500</span>) , <span class='fl'>2</span>]

<span class='fu'>PlotCategoryExpr</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
                 <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
                 <span class='kw'>test.stat</span>     <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                 <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"category-centered"</span>,
                 <span class='kw'>distr.type</span>    <span class='kw'>=</span> <span class='st'>"boxplot"</span>,
                 <span class='kw'>log.expr</span>      <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                 <span class='kw'>gene.set</span>      <span class='kw'>=</span> <span class='no'>example.gene.set</span>)

<span class='co'># }</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost, Alexander Gabel.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

