<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot Mean Relative Expression Levels as Barplot — PlotBarRE • myTAI</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Plot Mean Relative Expression Levels as Barplot — PlotBarRE" />

<meta property="og:description" content="This function takes a PhyloExpressionSet or DivergenceExpressionSet object as input and computes
for two or more defined phylostratum (divergence stratum) classes the statistical significance of
the differences of mean relative expression of these two (or more) corresponding phylostratum (divergence stratum) classes.
As test-statistic, the function performs a nonparametric kruskal.test
 based on relative expression values stored within each defined phylostratum class." />
<meta name="twitter:card" content="summary" />


<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.8.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot Mean Relative Expression Levels as Barplot</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/PlotBarRE.R'><code>R/PlotBarRE.R</code></a></small>
    <div class="hidden name"><code>PlotBarRE.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function takes a PhyloExpressionSet or DivergenceExpressionSet object as input and computes
for two or more defined phylostratum (divergence stratum) classes the statistical significance of
the differences of mean relative expression of these two (or more) corresponding phylostratum (divergence stratum) classes.
As test-statistic, the function performs a nonparametric <code>kruskal.test</code>
 based on relative expression values stored within each defined phylostratum class.</p>
    
    </div>

    <pre class="usage"><span class='fu'>PlotBarRE</span>(<span class='no'>ExpressionSet</span>, <span class='kw'>Groups</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>wLength</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>ratio</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>p.adjust.method</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>Groups</th>
      <td><p>a list containing the phylostrata or divergence-strata that correspond to the same phylostratum class or divergence class.
For ex. evolutionary old phylostrata: PS1-3 (Class 1) and evolutionary young phylostrata: PS4-12 (Class 2). 
In this case, the <code>Groups</code> list could be assigned as, <code>Groups</code> = list(c(1:3), c(4:12)). 
It is also possible to define more than 2 groups of evolutionary ages.
For ex. <code>Groups</code> = list(c(1:3),c(4:8),c(9:12)) would perform a <code>kruskal.test</code> to determine the statistical significance
of the evolutionary classes PS1-3, PS4-6, and PS9-12 based on their corresponding mean relative expression levels.</p></td>
    </tr>
    <tr>
      <th>wLength</th>
      <td><p>a numeric value defining the whiskers length above the bars. In case there are numerous different phylostratum classes
a smaller <code>wLength</code> parameter should be used for better visualizations.</p></td>
    </tr>
    <tr>
      <th>ratio</th>
      <td><p>a boolean value specifying whether the bars in the barplot represent the 
mean relative expression level of phylostrata belonging to the same phylostratum class.
In case <code>ratio</code> = TRUE, the ratio of the mean relative expression level of 
the two phylostrata classes is plotted as lines within the barplot. This parameter can only be used for 2 class comparisons.</p></td>
    </tr>
    <tr>
      <th>p.adjust.method</th>
      <td><p>correction method to adjust p-values for multiple comparisons (see <code>p.adjust</code> for possible methods).
E.g., <code>p.adjust.method = "BH"</code> (Benjamini &amp; Hochberg (1995)) or <code>p.adjust.method = "bonferroni"</code> (Bonferroni correction).</p></td>
    </tr>
    <tr>
      <th>&#8230;</th>
      <td><p>default graphics parameters.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A barplot comparing Phylostratum-Classes by its mean relative expression levels.
Significant stages are marked by '*' referring to statistically significant differences:</p>       
<p>(1) '*'   = P-Value &lt;= 0.05</p>
<p>(2) '**'  = P-Value &lt;= 0.005</p>
<p>(3) '***' = P-Value &lt;= 0.0005</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>In case a large number of developmental stages is included in the input <code>ExpressionSet</code>,
 p-values returned by <code>PlotBarRE</code> should be adjusted for multiple comparisons which can be done
 by specifying the <code>p.adjust.method</code> argument.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Quint M et al. 2012. "A transcriptomic hourglass in plant embryogenesis". Nature (490): 98-101.</p>
<p>Domazet-Loso T. and Tautz D. 2010. "A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns". Nature (468): 815-818.</p>
<p>Myles Hollander and Douglas A. Wolfe (1973), Nonparametric Statistical Methods. New York: John Wiley &amp; Sons. Pages 115-120.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='RE.html'>RE</a></code>, <code><a href='REMatrix.html'>REMatrix</a></code>,<code><a href='PlotRE.html'>PlotRE</a></code>, <code>kruskal.test</code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># read standard phylotranscriptomics data</span>
<span class='fu'>data</span>(<span class='no'>PhyloExpressionSetExample</span>)
<span class='fu'>data</span>(<span class='no'>DivergenceExpressionSetExample</span>)

<span class='co'># example PhyloExpressionSet</span>
<span class='fu'>PlotBarRE</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
          <span class='kw'>Groups</span>        <span class='kw'>=</span> <span class='fu'>list</span>(<span class='fu'>c</span>(<span class='fl'>1</span>:<span class='fl'>3</span>), <span class='fu'>c</span>(<span class='fl'>4</span>:<span class='fl'>12</span>)))</div><div class='img'><img src='PlotBarRE-1.png' alt='' width='700' height='433' /></div><div class='input'>

<span class='co'># example DivergenceExpressionSet</span>
<span class='fu'>PlotBarRE</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>DivergenceExpressionSetExample</span>,
          <span class='kw'>Groups</span>        <span class='kw'>=</span> <span class='fu'>list</span>(<span class='fu'>c</span>(<span class='fl'>1</span>:<span class='fl'>5</span>), <span class='fu'>c</span>(<span class='fl'>6</span>:<span class='fl'>10</span>)))</div><div class='img'><img src='PlotBarRE-2.png' alt='' width='700' height='433' /></div><div class='input'>

<span class='co'># Perform PlotBarRE() with p-value adjustment method Benjamini &amp; Hochberg (1995)</span>
<span class='fu'>PlotBarRE</span>(<span class='kw'>ExpressionSet</span>   <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
          <span class='kw'>Groups</span>          <span class='kw'>=</span> <span class='fu'>list</span>(<span class='fu'>c</span>(<span class='fl'>1</span>:<span class='fl'>3</span>), <span class='fu'>c</span>(<span class='fl'>4</span>:<span class='fl'>12</span>)),
          <span class='kw'>p.adjust.method</span> <span class='kw'>=</span> <span class='st'>"BH"</span>)</div><div class='img'><img src='PlotBarRE-3.png' alt='' width='700' height='433' /></div><div class='input'>

<span class='co'># Example: plot ratio</span>
<span class='co'># the ratio curve visualizes the ratio between bar 1 / bar 2</span>
<span class='co'># the z - axis shows the corresponding ratio value of bar 1 / bar 2</span>
<span class='fu'>PlotBarRE</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
          <span class='kw'>Groups</span>        <span class='kw'>=</span> <span class='fu'>list</span>(<span class='fu'>c</span>(<span class='fl'>1</span>:<span class='fl'>3</span>), <span class='fu'>c</span>(<span class='fl'>4</span>:<span class='fl'>12</span>)),
          <span class='kw'>ratio</span>         <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='img'><img src='PlotBarRE-4.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Hajk-Georg Drost

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

