<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot the significant differences between gene expression distributions of PS or DS groups — PlotGroupDiffs • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Plot the significant differences between gene expression distributions of PS or DS groups — PlotGroupDiffs" />

<meta property="og:description" content="This function performs a test to quantify the statistical significance between
the global expression level distributions of groups of PS or DS. It therefore, allows users to investigate
significant groups of PS or DS that significantly differ in their gene expression level distibution
within specific developmental stages or experiments." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the significant differences between gene expression distributions of PS or DS groups</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/PlotGroupDiffs.R'><code>R/PlotGroupDiffs.R</code></a></small>
    <div class="hidden name"><code>PlotGroupDiffs.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function performs a test to quantify the statistical significance between
the global expression level distributions of groups of PS or DS. It therefore, allows users to investigate
significant groups of PS or DS that significantly differ in their gene expression level distibution
within specific developmental stages or experiments.</p>
    
    </div>

    <pre class="usage"><span class='fu'>PlotGroupDiffs</span>(<span class='no'>ExpressionSet</span>, <span class='kw'>Groups</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>legendName</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>stat.test</span> <span class='kw'>=</span> <span class='st'>"wilcox.test"</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"turquoise3"</span>, <span class='st'>"magenta3"</span>),
  <span class='kw'>plot.type</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>gene.set</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>Groups</th>
      <td><p>a list containing the phylostrata or divergence strata that correspond 
to the same phylostratum class or divergence class.
For ex. evolutionary old phylostrata: PS1-3 (Class 1) 
and evolutionary young phylostrata: PS4-12 (Class 2). In this case, 
the list could be assigned as, <code>Groups</code> = list(c(1:3), c(4:12)).</p></td>
    </tr>
    <tr>
      <th>legendName</th>
      <td><p>a character string specifying whether "PS" or "DS" are used to compute relative expression profiles.</p></td>
    </tr>
    <tr>
      <th>stat.test</th>
      <td><p>the statistical test to quantify PS or DS group differences.</p></td>
    </tr>
    <tr>
      <th>col</th>
      <td><p>colors for the two box plots representing the expression level distributions of selected PS/DS groups.</p></td>
    </tr>
    <tr>
      <th>plot.type</th>
      <td><p>the type of plot that shall be drawn to visualized the difference in PS/DS group specific expression .</p></td>
    </tr>
    <tr>
      <th>gene.set</th>
      <td><p>a character vector storing the gene ids for which group specific differences shall be statistically quantified.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>additional plot parameters.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The purpose of this function is to detect groups of PS or DS that significantly differ in their gene expression
level distributions on a global (transcriptome) level. Since relative expression levels (<code><a href='PlotRE.html'>PlotRE</a></code>) or
PS or DS specific mean expression levels (<code><a href='PlotMeans.html'>PlotMeans</a></code>) are biased by highly expressed genes,
this function allows users to objectively test the significant difference of transcriptome expression between
groups of PS or DS in a specific developmental stage or experiment.</p>
<p>In particular, this function divides (for each developmental stage separately) the gene expression levels into two groups: Group1 = genes deriving from selected PS/DS in group 1 and 
Group2 = genes deriving from selected PS/DS in group 2. Within each stage the expression level distributions between group 1 and group 2 are statistically quantified using a <code><a href='https://www.rdocumentation.org/packages/stats/topics/wilcox.test'>wilcox.test</a></code>.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='PlotMeans.html'>PlotMeans</a></code>, <code><a href='PlotRE.html'>PlotRE</a></code>, <code><a href='PlotBarRE.html'>PlotBarRE</a></code>, <code><a href='PlotCategoryExpr.html'>PlotCategoryExpr</a></code>, <code><a href='GroupDiffs.html'>GroupDiffs</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>PhyloExpressionSetExample</span>)

<span class='fu'>PlotGroupDiffs</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
               <span class='kw'>Groups</span>        <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='kw'>group_1</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>3</span>,<span class='kw'>group_2</span> <span class='kw'>=</span> <span class='fl'>4</span>:<span class='fl'>12</span>),
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"b"</span>,
               <span class='kw'>lwd</span>           <span class='kw'>=</span> <span class='fl'>6</span>,
               <span class='kw'>xlab</span>          <span class='kw'>=</span> <span class='st'>"Ontogeny"</span>)</div><div class='output co'>#&gt;                               Zygote     Quadrant     Globular        Heart
#&gt; p.value ( wilcox.test ) 3.362424e-34 3.017793e-42 1.987181e-71 1.546567e-66
#&gt;                              Torpedo         Bent       Mature
#&gt; p.value ( wilcox.test ) 1.174961e-85 9.975191e-90 1.007102e-74</div><div class='input'>

<span class='co'># only receive the p-values without the corresponding plot               </span>
<span class='fu'>PlotGroupDiffs</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
               <span class='kw'>Groups</span>        <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='kw'>group_1</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>3</span>,<span class='kw'>group_2</span> <span class='kw'>=</span> <span class='fl'>4</span>:<span class='fl'>12</span>),
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>plot.p.vals</span>   <span class='kw'>=</span> <span class='fl'>FALSE</span>,
               <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"b"</span>,
               <span class='kw'>lwd</span>           <span class='kw'>=</span> <span class='fl'>6</span>,
               <span class='kw'>xlab</span>          <span class='kw'>=</span> <span class='st'>"Ontogeny"</span>)</div><div class='output co'>#&gt;                               Zygote     Quadrant     Globular        Heart
#&gt; p.value ( wilcox.test ) 3.362424e-34 3.017793e-42 1.987181e-71 1.546567e-66
#&gt;                              Torpedo         Bent       Mature
#&gt; p.value ( wilcox.test ) 1.174961e-85 9.975191e-90 1.007102e-74</div><div class='input'>

<span class='co'># quantify the significant difference of a selected set of genes</span>
<span class='co'># only receive the p-values without the corresponding plot</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/Random'>set.seed</a></span>(<span class='fl'>123</span>)
<span class='no'>ExampleGeneSet</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/sample'>sample</a></span>(<span class='no'>PhyloExpressionSetExample</span>[ , <span class='fl'>2</span>],<span class='fl'>5000</span>)

<span class='fu'>PlotGroupDiffs</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='no'>PhyloExpressionSetExample</span>,
               <span class='kw'>Groups</span>        <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='kw'>group_1</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>3</span>,<span class='kw'>group_2</span> <span class='kw'>=</span> <span class='fl'>4</span>:<span class='fl'>12</span>),
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>plot.p.vals</span>   <span class='kw'>=</span> <span class='fl'>FALSE</span>,
               <span class='kw'>gene.set</span>      <span class='kw'>=</span> <span class='no'>ExampleGeneSet</span>)</div><div class='output co'>#&gt;                               Zygote     Quadrant     Globular       Heart
#&gt; p.value ( wilcox.test ) 4.314214e-06 3.669484e-07 2.606245e-10 8.99476e-09
#&gt;                            Torpedo         Bent       Mature
#&gt; p.value ( wilcox.test ) 5.2788e-11 5.998163e-12 1.827263e-14</div><div class='input'>

<span class='co'># plot differences as boxplot for each developmental stage</span>
<span class='fu'>PlotGroupDiffs</span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='fu'><a href='tf.html'>tf</a></span>(<span class='no'>PhyloExpressionSetExample</span>,<span class='no'>log2</span>),
               <span class='kw'>Groups</span>        <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='kw'>group_1</span> <span class='kw'>=</span> <span class='fl'>1</span>:<span class='fl'>3</span>,<span class='kw'>group_2</span> <span class='kw'>=</span> <span class='fl'>4</span>:<span class='fl'>12</span>),
               <span class='kw'>legendName</span>    <span class='kw'>=</span> <span class='st'>"PS"</span>,
               <span class='kw'>plot.type</span>     <span class='kw'>=</span> <span class='st'>"boxplot"</span>)</div><div class='output co'>#&gt;                               Zygote     Quadrant     Globular        Heart
#&gt; p.value ( wilcox.test ) 3.362424e-34 3.017793e-42 1.987181e-71 1.546567e-66
#&gt;                              Torpedo         Bent       Mature
#&gt; p.value ( wilcox.test ) 1.174961e-85 9.975191e-90 1.007102e-74</div><div class='input'>

</div><div class='img'><img src='PlotGroupDiffs-1.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#details">Details</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

