<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Transform Gene Expression Levels — tf • myTAI</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Transform Gene Expression Levels — tf" />

<meta property="og:description" content="This function transforms the gene expression set stored in an input 
PhloExpressionSet or DivergenceExpressionSet object and returns a 
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels. 
The resulting transformed PhloExpressionSet or DivergenceExpressionSet 
object can then be used for subsequent analyses based on transformed expression levels." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.8.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI Package</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HajkD/myTAI">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Transform Gene Expression Levels</h1>
    <small class="dont-index">Source: <a href='https://github.com/HajkD/myTAI/blob/master/R/tf.R'><code>R/tf.R</code></a></small>
    <div class="hidden name"><code>tf.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function transforms the gene expression set stored in an input 
PhloExpressionSet or DivergenceExpressionSet object and returns a 
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels. 
The resulting transformed PhloExpressionSet or DivergenceExpressionSet 
object can then be used for subsequent analyses based on transformed expression levels.</p>
    
    </div>

    <pre class="usage"><span class='fu'>tf</span>(<span class='no'>ExpressionSet</span>, <span class='no'>FUN</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>FUN</th>
      <td><p>any valid function that transformes gene expression levels.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a standard PhloExpressionSet or DivergenceExpressionSet object storing transformed gene expression levels.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Motivated by the dicussion raised by Piasecka et al., 2013, the influence of
gene expression transformation on the global phylotranscriptomics pattern does not seem negligible.
Hence, different transformations can result in qualitatively different <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code>
patterns.</p>
<p>Initially, the <code><a href='TAI.html'>TAI</a></code> and <code><a href='TDI.html'>TDI</a></code> formulas were defined for absolute expression levels.
So using the initial <code><a href='TAI.html'>TAI</a></code> and <code><a href='TDI.html'>TDI</a></code> formulas with transformed expression levels
might turn out in qualitatively different patterns when compared with non-transformed expression levels,
but might also belong to a different class of models, since different valid expression level transformation functions result in different patterns.</p>
<p>The purpose of this function is to allow the user to study the qualitative impact of different transformation functions on 
the global <code><a href='TAI.html'>TAI</a></code> and <code><a href='TDI.html'>TDI</a></code> pattern, or on any subsequent phylotranscriptomics analysis.</p>
<p>The examples using the <em>PhyloExpressionSetExample</em> data set show that using common gene expression 
transformation functions: <code><a href='https://www.rdocumentation.org/packages/base/topics/Log'>log2</a></code> (Quackenbush, 2001 and 2002), <code><a href='https://www.rdocumentation.org/packages/base/topics/MathFun'>sqrt</a></code> (Yeung et al., 2001), 
<code><a href='https://www.rdocumentation.org/packages/MASS/topics/boxcox'>boxcox</a></code>, or <em>inverse hyperbolic sine transformation</em>, each transformation results 
in qualitatively different patterns. 
Nevertheless, for each resulting pattern the statistical significance can be tested 
using either the <code><a href='FlatLineTest.html'>FlatLineTest</a></code> or <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code> (Drost et al., 2014) 
to quantify the significance of interest.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Piasecka B, Lichocki P, Moretti S, et al. (2013) The hourglass and the early conservation models--co-existing patterns of developmental constraints in vertebrates. PLoS Genet. 9(4): e1003476.</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.</p>
<p>Domazet-Loso T., Tautz D. (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.</p>
<p>Drost H.G., Gabel A., Grosse I., Quint M. (2015) Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis. Mol Biol Evol. 32 (5): 1221-1231. doi: 10.1093/molbev/msv012</p>
<p>K.Y. Yeung et al.: Model-based clustering and data transformations for gene expression data. Bioinformatics 2001, 17:977-987</p>
<p>K.Y. Yeung et al.:  Supplement to Model-based clustering and data transformations for gene expression data - Data Transformations and the Gaussian mixture assumption. Bioinformatics 2001, 17:977-987</p>
<p>P.A.C. Hoen et al.: Deep sequencing-based expression analysis shows major advances in robustness, resolution and inter-lab portability over five microarray platforms. Nucleic Acids Research 2008, Vol. 36, No. 21</p>
<p>H.H. Thygesen et al.: Comparing transformation methods for DNA microarray data. BMC Bioinformatics 2004, 5:77</p>
<p>John Quackenbush: Microarray data normalization and transformation. Nature Genetics 2002, 32:496-501</p>
<p>John Quackenbush: Computational Analysis of Microarray Data. Nature Reviews 2001, 2:418-427</p>
<p>R. Nadon and J. Shoemaker: Statistical issues with microarrays: processing and analysis. TRENDS in Genetics 2002, Vol. 18 No. 5:265-271</p>
<p>B.P. Durbin et al.: A variance-stabilizing transformation for gene-expression microarray data. Bioinformatics 2002, 18:S105-S110</p>
<p>J. M. Bland et al.: Transforming data. BMJ 1996, 312:770</p>
<p><a href='http://stats.stackexchange.com/questions/1444/how-should-i-transform-non-negative-data-including-zeros'>http://stats.stackexchange.com/questions/1444/how-should-i-transform-non-negative-data-including-zeros</a></p>
<p><a href='http://stats.stackexchange.com/questions/78929/how-can-i-estimate-theta-for-the-inverse-hyperbolic-sine-transformation'>http://stats.stackexchange.com/questions/78929/how-can-i-estimate-theta-for-the-inverse-hyperbolic-sine-transformation</a></p>
<p>John B. Burbidge, Lonnie Magee and A. Leslie Robb (1988) Alternative Transformations to Handle Extreme Values of the Dependent Variable. Journal of the American Statistical Association, 83(401): 123-127.</p>
<p>G. E. P. Box and D. R. Cox (1964) An Analysis of Transformations. Journal of the Royal Statistical Society. Series B (Methodological), 26(2): 211-252.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='TAI.html'>TAI</a></code>, <code><a href='TDI.html'>TDI</a></code>, <code><a href='FlatLineTest.html'>FlatLineTest</a></code>, <code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>PhyloExpressionSetExample</span>)

<span class='co'># a simple example is to transform the gene expression levels</span>
<span class='co'># of a given PhyloExpressionSet using a sqrt or log2 transformation</span>

<span class='no'>PES.sqrt</span> <span class='kw'>&lt;-</span> <span class='fu'>tf</span>(<span class='no'>PhyloExpressionSetExample</span>, <span class='no'>sqrt</span>)

<span class='no'>PES.log2</span> <span class='kw'>&lt;-</span> <span class='fu'>tf</span>(<span class='no'>PhyloExpressionSetExample</span>, <span class='no'>log2</span>)

<span class='co'># in case a given PhyloExpressionSet already stores gene expression levels</span>
<span class='co'># that are log2 transformed and need to be re-transformed to absolute</span>
<span class='co'># expression levels, to perform subsequent phylotranscriptomics analyses </span>
<span class='co'># (that are defined for absolute expression levels), one can re-transform</span>
<span class='co'># a PhyloExpressionSet like this:</span>

<span class='no'>PES.absolute</span> <span class='kw'>&lt;-</span> <span class='fu'>tf</span>(<span class='no'>PES.log2</span> , <span class='kw'>function</span>(<span class='no'>x</span>) <span class='fl'>2</span>^<span class='no'>x</span>)

<span class='co'># which should be the same as  PhyloExpressionSetExample :</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='no'>PhyloExpressionSetExample</span>)</div><div class='output co'>#&gt;   Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart
#&gt; 1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224
#&gt; 2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612
#&gt; 3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195
#&gt; 4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734
#&gt; 5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405
#&gt; 6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180
#&gt;      Torpedo       Bent    Mature
#&gt; 1  1000.2529   962.9772 1696.4274
#&gt; 2  1496.3207  1114.6435 1071.6555
#&gt; 3   864.2180   877.2060  894.8189
#&gt; 4  1392.6429  1287.9746  861.2605
#&gt; 5 56231.5689 66980.3673 7772.5617
#&gt; 6   942.8453   870.2625  792.7542</div><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='no'>PES.absolute</span>)</div><div class='output co'>#&gt;   Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart
#&gt; 1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224
#&gt; 2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612
#&gt; 3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195
#&gt; 4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734
#&gt; 5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405
#&gt; 6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180
#&gt;      Torpedo       Bent    Mature
#&gt; 1  1000.2529   962.9772 1696.4274
#&gt; 2  1496.3207  1114.6435 1071.6555
#&gt; 3   864.2180   877.2060  894.8189
#&gt; 4  1392.6429  1287.9746  861.2605
#&gt; 5 56231.5689 66980.3673 7772.5617
#&gt; 6   942.8453   870.2625  792.7542</div><div class='input'>

<span class='co'># plotting the TAI using log2 transformed expression levels</span>
<span class='co'># and performing the Flat Line Test to obtain the p-value</span>
<span class='fu'><a href='PlotPattern.html'>PlotPattern</a></span>(<span class='kw'>ExpressionSet</span> <span class='kw'>=</span> <span class='fu'>tf</span>(<span class='no'>PhyloExpressionSetExample</span>, <span class='no'>log2</span>),
            <span class='kw'>type</span>          <span class='kw'>=</span> <span class='st'>"l"</span>,
            <span class='kw'>lwd</span>           <span class='kw'>=</span> <span class='fl'>5</span>,
            <span class='kw'>TestStatistic</span> <span class='kw'>=</span> <span class='st'>"FlatLineTest"</span>)</div><div class='img'><img src='tf-1.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.2.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

